
```{r, include = FALSE} 
library(ggplot2)#7 rows
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }
load("~/Documents/project/normaliseddataCPE.Rdata")
load("~/Documents/project/modelmatricesCPE.Rdata")
#load(file="~/Documents/project/normaliseddata.Rdata")
HER2Basal <- dataHER2Basal
#load(file="~/Documents/project/modelmatrices.Rdata")

```






read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XHER2Basal <- model.matrix(~ subtype * CPE, data=as.data.frame(model_HER2Basal))
fitHER2Basal <- lmFit(as.data.frame(HER2Basal), XHER2Basal)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XHER2Basal))
#contr.matrix
#contrastsfitHER2Basal <- contrasts.fit(fitHER2Basal, contrasts=contr.matrix)
#contrastsfitHER2Basal <- eBayes(contrastsfitHER2Basal)
fitHER2Basal <- eBayes(fitHER2Basal)
TTHER2Basal <- topTable(fitHER2Basal, number = dim(HER2Basal)[1])
pvalsHER2Basal <- fitHER2Basal$p.value
fdradjusted_pvalsHER2Basal <- p.adjust(pvalsHER2Basal[,4], method = "fdr")
fdradjusted_pvalsHER2Basalsig <- fdradjusted_pvalsHER2Basal[which(fdradjusted_pvalsHER2Basal < 0.05)]
names(fdradjusted_pvalsHER2Basalsig)
top95HER2Basal <- TTHER2Basal[which(rownames(TTHER2Basal) %in% names(fdradjusted_pvalsHER2Basalsig)),]
#plotMA(treatedTT4HER2Basal , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTHER2Basal$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitHER2Basal, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalsHER2Basal)
treatedTT1HER2Basal <- topTreat(fitHER2Basal, coef=1, number=dim(HER2Basal)[1], lfc=1)
treatedTT1HER2Basalsig <- treatedTT1HER2Basal[which(treatedTT1HER2Basal$adj.P.Val < 0.05),]
treatedTT1HER2Basalsig
treatedTT2HER2Basal <- topTreat(fitHER2Basal, coef=2, number=dim(HER2Basal)[1], lfc=1)
treatedTT2HER2Basalsig <- treatedTT2HER2Basal[which(treatedTT2HER2Basal$adj.P.Val < 0.05),]
treatedTT2HER2Basalsig
treatedTT3HER2Basal <- topTreat(fitHER2Basal, coef=3, number=dim(HER2Basal)[1], lfc=1)
treatedTT3HER2Basalsig <- treatedTT3HER2Basal[which(treatedTT3HER2Basal$adj.P.Val < 0.05),]
treatedTT3HER2Basalsig
treatedTT4HER2Basal <- topTreat(fitHER2Basal, coef=4, number=dim(HER2Basal)[1], lfc=1)
treatedTT4HER2Basalsig <- treatedTT4HER2Basal[which(treatedTT4HER2Basal$adj.P.Val < 0.05),]
treatedTT4HER2Basalsig
#plot(density(treatedTT2HER2Basalsig$logFC))
#ENSG00000122566
#ENSG00000122566

save(treatedTT4HER2Basalsig, treatedTT2HER2Basalsig, file="HER2BasaloutputCPE.Rdata")
```






```{r}
plot(treatedTT4HER2Basal$logFC, -log10(treatedTT4HER2Basal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4HER2Basalsig$logFC, -log10(treatedTT4HER2Basalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_HER2Basal <-model_HER2Basal
nmodel_HER2Basal$CPE <- (1- nmodel_HER2Basal$CPE)
nXHER2Basal <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_HER2Basal))
nfitHER2Basal <- lmFit(as.data.frame(HER2Basal), nXHER2Basal)

nfitHER2Basal <- eBayes(nfitHER2Basal)
nTTHER2Basal<- topTable(nfitHER2Basal, number = dim(HER2Basal)[1])
npvalsHER2Basal <- nfitHER2Basal$p.value
nfdradjusted_pvalsHER2Basal <- p.adjust(npvalsHER2Basal[,4], method = "fdr")
nfdradjusted_pvalsHER2Basalsig <- nfdradjusted_pvalsHER2Basal[which(nfdradjusted_pvalsHER2Basal < 0.05)]
names(nfdradjusted_pvalsHER2Basalsig)
ntop95HER2Basal <- nTTHER2Basal[which(rownames(nTTHER2Basal) %in% names(nfdradjusted_pvalsHER2Basalsig)),]

ntreatedTT1HER2Basal <- topTreat(nfitHER2Basal, coef=1, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT1HER2Basalsig <- ntreatedTT1HER2Basal[which(ntreatedTT1HER2Basal$adj.P.Val < 0.05),]
ntreatedTT1HER2Basalsig
ntreatedTT2HER2Basal <- topTreat(nfitHER2Basal, coef=2, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT2HER2Basalsig <- ntreatedTT2HER2Basal[which(ntreatedTT2HER2Basal$adj.P.Val < 0.05),]
ntreatedTT2HER2Basalsig
ntreatedTT3HER2Basal <- topTreat(nfitHER2Basal, coef=3, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT3HER2Basalsig <- ntreatedTT3HER2Basal[which(ntreatedTT3HER2Basal$adj.P.Val < 0.05),]
ntreatedTT3HER2Basalsig
ntreatedTT4HER2Basal <- topTreat(nfitHER2Basal, coef=4, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT4HER2Basalsig <- ntreatedTT4HER2Basal[which(ntreatedTT4HER2Basal$adj.P.Val < 0.05),]
ntreatedTT4HER2Basalsig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_HER2Basal <- as.data.frame(model_HER2Basal)
nCPEmodel_HER2Basal$CPE <- (nCPEmodel_HER2Basal$CPE)
nthelm2HER2Basal <- lm(HER2Basal[17876,] ~ subtype * CPE, data=nCPEmodel_HER2Basal)

plot_model(nthelm2HER2Basal, type = "pred", terms = c( "CPE","subtype"))
which(rownames(HER2Basal) == "ENSG00000214688") # this is a gene responsible for ER expression
```

```{r}
normalHER2Basal <- treatedTT2HER2Basalsig[which(rownames(treatedTT2HER2Basalsig) %in% rownames(treatedTT4HER2Basalsig)),]
normalHER2Basal <- normalHER2Basal[which(!rownames(normalHER2Basal) %in% rownames(ntreatedTT2HER2Basalsig)),]

cancerHER2Basal <- ntreatedTT2HER2Basalsig[which(rownames(ntreatedTT2HER2Basalsig) %in% rownames(ntreatedTT4HER2Basalsig)),]
cancerHER2Basal <- cancerHER2Basal[which(!rownames(cancerHER2Basal) %in% rownames(treatedTT2HER2Basalsig)),]


save(cancerHER2Basal, normalHER2Basal, file="HER2Basal_C_N_CPE.Rdata")
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4HER2Basalsig$status <- ifelse(rownames(treatedTT4HER2Basalsig) %in% rownames(cancerHER2Basal), "Within Cancer Cells", ifelse(rownames(treatedTT4HER2Basalsig) %in% rownames(normalHER2Basal), "Within Normal Cells", "Both"))
treatedTT4HER2Basalsig$colour <- ifelse(rownames(treatedTT4HER2Basalsig) %in% rownames(cancerHER2Basal), cb3[2], ifelse(rownames(treatedTT4HER2Basalsig) %in% rownames(normalHER2Basal), cb3[3], cb3[1]))
genesHER2Basal <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4HER2Basalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genesHER2Basal <- genesHER2Basal[!duplicated(genesHER2Basal$GENEID),]
genesHER2Basal$logFC <- treatedTT4HER2Basalsig[which(rownames(treatedTT4HER2Basalsig) %in% genesHER2Basal$GENEID),][genesHER2Basal$GENEID,]$logFC
genesHER2Basal$status <- treatedTT4HER2Basalsig[which(rownames(treatedTT4HER2Basalsig) %in% genesHER2Basal$GENEID),][genesHER2Basal$GENEID,]$status
HER2Basalsigs <- ggplot(genesHER2Basal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("HER2-enriched vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("HER2BasalsigsCPE.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlyHER2Basal <-genesHER2Basal[which(genesHER2Basal$status != "Both"),]
cancernormalonlyHER2Basalgraph <- ggplot(cancernormalonlyHER2Basal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("HER2-enriched vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlyHER2BasalCPE.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XHER2Basals <- model.matrix(~ subtype + CPE, data=as.data.frame(model_HER2Basal))
fitHER2Basals <- lmFit(as.data.frame(HER2Basal), XHER2Basals)
fitHER2Basals <- eBayes(fitHER2Basals)
TTHER2Basals <- topTable(fitHER2Basals, number = dim(HER2Basal)[1])
pvalsHER2Basals <- fitHER2Basals$p.value
fdradjusted_pvalsHER2BasalINTERs <- p.adjust(pvalsHER2Basals[,1], method = "fdr")
fdradjusted_pvalsHER2BasalsigINTERs <- fdradjusted_pvalsHER2BasalINTERs[which(fdradjusted_pvalsHER2BasalINTERs < 0.05)]
length(names(fdradjusted_pvalsHER2BasalsigINTERs))
fdradjusted_pvalsHER2BasalSUBs <- p.adjust(pvalsHER2Basals[,2], method = "fdr")
fdradjusted_pvalsHER2BasalsigSUBs <- fdradjusted_pvalsHER2BasalSUBs[which(fdradjusted_pvalsHER2BasalSUBs < 0.05)]
length(names(fdradjusted_pvalsHER2BasalsigSUBs))
fdradjusted_pvalsHER2BasalCPEs <- p.adjust(pvalsHER2Basals[,3], method = "fdr")
fdradjusted_pvalsHER2BasalsigCPEs <- fdradjusted_pvalsHER2BasalCPEs[which(fdradjusted_pvalsHER2BasalCPEs < 0.05)]
length(names(fdradjusted_pvalsHER2BasalsigCPEs))
plotSA(fitHER2Basals, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalsHER2BasalSUBs)

treatedTT1HER2Basals <- topTreat(fitHER2Basals, coef=1, number=dim(HER2Basal)[1], lfc=1)
treatedTT1HER2Basalsigs <- treatedTT1HER2Basals[which(treatedTT1HER2Basals$adj.P.Val < 0.05),]
treatedTT1HER2Basalsigs
treatedTT2HER2Basals <- topTreat(fitHER2Basals, coef=2, number=dim(HER2Basal)[1], lfc=1)
treatedTT2HER2Basalsigs <- treatedTT2HER2Basals[which(treatedTT2HER2Basals$adj.P.Val < 0.05),]
treatedTT2HER2Basalsigs
treatedTT3HER2Basals <- topTreat(fitHER2Basals, coef=3, number=dim(HER2Basal)[1], lfc=1)
treatedTT3HER2Basalsigs <- treatedTT3HER2Basals[which(treatedTT3HER2Basals$adj.P.Val < 0.05),]
treatedTT3HER2Basalsigs


treatedTT2HER2Basalsigs[which(rownames(treatedTT2HER2Basalsigs) %in% rownames(treatedTT4HER2Basalsig)),]
treatedTT2HER2Basalsigs[which(rownames(treatedTT2HER2Basalsigs) %in% rownames(cancerHER2Basal)),]
treatedTT2HER2Basalsigs[which(rownames(treatedTT2HER2Basalsigs) %in% rownames(normalHER2Basal)),]
```

```{r}
plot(treatedTT4HER2Basal$logFC, -log10(treatedTT4HER2Basal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4HER2Basalsig$logFC, -log10(treatedTT4HER2Basalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_HER2Basals <- model_HER2Basal
nmodel_HER2Basals$CPE <- (1- nmodel_HER2Basals$CPE)
nXHER2Basals <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_HER2Basals))
nfitHER2Basals <- lmFit(as.data.frame(HER2Basal), nXHER2Basals)
nfitHER2Basals <- eBayes(nfitHER2Basals)
nTTHER2Basals<- topTable(nfitHER2Basals, number = dim(HER2Basal)[1])
npvalsHER2Basals <- nfitHER2Basals$p.value
ntreatedTT1HER2Basals <- topTreat(nfitHER2Basals, coef=1, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT1HER2Basalsigs <- ntreatedTT1HER2Basals[which(ntreatedTT1HER2Basals$adj.P.Val < 0.05),]
ntreatedTT1HER2Basalsigs
ntreatedTT2HER2Basals <- topTreat(nfitHER2Basals, coef=2, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT2HER2Basalsigs <- ntreatedTT2HER2Basals[which(ntreatedTT2HER2Basals$adj.P.Val < 0.05),]
ntreatedTT2HER2Basalsigs
#ntreatedTT2HER2Basalsigs[which(rownames(ntreatedTT2HER2Basalsigs) %in% rownames(normalHER2Basal)),]
ntreatedTT3HER2Basals <- topTreat(nfitHER2Basals, coef=3, number=dim(HER2Basal)[1], lfc=1)
ntreatedTT3HER2Basalsigs <- ntreatedTT3HER2Basals[which(ntreatedTT3HER2Basals$adj.P.Val < 0.05),]
ntreatedTT3HER2Basalsigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_HER2Basals <- as.data.frame(nmodel_HER2Basals)
nCPEmodel_HER2Basals$CPE <- (1- nCPEmodel_HER2Basals$CPE)
nthelm2HER2Basals <- lm(HER2Basal[5015,] ~ subtype + CPE, data=nCPEmodel_HER2Basals)

plot_model(nthelm2HER2Basals, type = "pred", terms = c( "CPE","subtype"))
which(rownames(HER2Basal) == "ENSG00000122566")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2HER2Basals$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2HER2Basals$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("HER2-enriched", "Basal-like")
CPEmodel_HER2Basal <- as.data.frame(model_HER2Basal)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1HER2Basalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GenesHER2Basal <- HER2Basal[which(rownames(HER2Basal) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GenesHER2Basal[top10Genes$GENEID,]
top10GenesHER2Basal <- top10GenesHER2Basal[order(rownames(top10GenesHER2Basal)),]
dim(top10GenesHER2Basal)
all.equal(rownames(top10GenesHER2Basal), top10Genes$GENEID)
rownames(top10GenesHER2Basal) <- top10Genes$SYMBOL
top10lmHER2Basal <-  numeric(length = nrow(top10GenesHER2Basal))
plotHER2BasalCPE1 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE1, file=paste0("~/Documents/project/plots/HER2BasalCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotHER2BasalCPE2 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE2, file=paste0("~/Documents/project/plots/HER2BasalCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotHER2BasalCPE3 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE3, file=paste0("~/Documents/project/plots/HER2BasalCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotHER2BasalCPE4 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE4, file=paste0("~/Documents/project/plots/HER2BasalCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotHER2BasalCPE5 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE5, file=paste0("~/Documents/project/plots/HER2BasalCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotHER2BasalCPE6 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE6, file=paste0("~/Documents/project/plots/HER2BasalCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotHER2BasalCPE7 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE7, file=paste0("~/Documents/project/plots/HER2BasalCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotHER2BasalCPE8 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE8, file=paste0("~/Documents/project/plots/HER2BasalCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotHER2BasalCPE9 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GenesHER2Basal)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE9, file=paste0("~/Documents/project/plots/HER2BasalCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotHER2BasalCPE10 <- ggplot(CPEmodel_HER2Basal ,aes(y=top10GenesHER2Basal[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GenesHER2Basal)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotHER2BasalCPE10, file=paste0("~/Documents/project/plots/HER2BasalCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPEHER2Basal <-cowplot::plot_grid(plotHER2BasalCPE1, plotHER2BasalCPE2 + theme(legend.position="none"), plotHER2BasalCPE3 + theme(legend.position="none"),plotHER2BasalCPE4 + theme(legend.position="none"), plotHER2BasalCPE5 + theme(legend.position="none"), plotHER2BasalCPE6 + theme(legend.position="none"),plotHER2BasalCPE7 + theme(legend.position="none"), plotHER2BasalCPE8 + theme(legend.position="none"), plotHER2BasalCPE9 + theme(legend.position="none"),plotHER2BasalCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPEHER2Basal, file=paste0("~/Documents/project/plots/mergedCPEHER2Basal.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XHER2Basalss <- model.matrix(~ subtype, data=as.data.frame(model_HER2Basal))
fitHER2Basalss <- lmFit(as.data.frame(HER2Basal), XHER2Basalss)

fitHER2Basalss <- eBayes(fitHER2Basalss)
TTHER2Basalss <- topTable(fitHER2Basalss, number = dim(HER2Basal)[1])
pvalsHER2Basalss <- fitHER2Basalss$p.value
fdradjusted_pvalsHER2BasalINTERss <- p.adjust(pvalsHER2Basalss[,1], method = "fdr")
fdradjusted_pvalsHER2BasalsigINTERss <- fdradjusted_pvalsHER2BasalINTERss[which(fdradjusted_pvalsHER2BasalINTERss < 0.05)]
length(names(fdradjusted_pvalsHER2BasalsigINTERss))
plotSA(fitHER2Basalss, main="Final model: Mean-variance trend")

treatedTT1HER2Basalss <- topTreat(fitHER2Basalss, coef=1, number=dim(HER2Basal)[1], lfc=1)
treatedTT1HER2Basalsigss <- treatedTT1HER2Basalss[which(treatedTT1HER2Basalss$adj.P.Val < 0.05),]
treatedTT1HER2Basalsigss
```

```{r}
plot(treatedTT1HER2Basalss$logFC, -log10(treatedTT1HER2Basalss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1HER2Basalsigss$logFC, -log10(treatedTT1HER2Basalsigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4HER2Basal$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4HER2Basal$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4HER2Basal$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3HER2Basal$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2HER2Basal$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1HER2Basal$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4HER2Basal$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3HER2Basal$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2HER2Basal$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1HER2Basal$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1HER2Basals$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1HER2Basals$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2HER2Basals$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2HER2Basals$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3HER2Basals$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3HER2Basals$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1HER2Basalsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1HER2Basals$adj.P.Val) / length(ntreatedTT1HER2Basals$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2HER2Basals$adj.P.Val) / length(ntreatedTT2HER2Basals$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3HER2Basals$adj.P.Val) / length(ntreatedTT3HER2Basals$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1HER2Basalss$adj.P.Val) / length(treatedTT1HER2Basalss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4HER2Basal$adj.P.Val) / length(ntreatedTT4HER2Basal$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2HER2Basal$adj.P.Val) / length(ntreatedTT2HER2Basal$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3HER2Basal$adj.P.Val) / length(ntreatedTT3HER2Basal$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1HER2Basal$adj.P.Val) / length(ntreatedTT1HER2Basal$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1HER2Basal$adj.P.Val) / length(ntreatedTT1HER2Basal$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4HER2Basal$adj.P.Val) / length(ntreatedTT4HER2Basal$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2HER2Basal$adj.P.Val) / length(ntreatedTT2HER2Basal$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3HER2Basal$adj.P.Val) / length(ntreatedTT3HER2Basal$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1HER2Basal$adj.P.Val) / length(treatedTT1HER2Basal$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3HER2Basal$adj.P.Val) / length(treatedTT3HER2Basal$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4HER2Basal$adj.P.Val) / length(treatedTT4HER2Basal$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2HER2Basal$adj.P.Val) / length(treatedTT2HER2Basal$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2HER2Basal$adj.P.Val) / length(treatedTT2HER2Basal$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2HER2Basal$adj.P.Val) / length(ntreatedTT2HER2Basal$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3HER2Basal$adj.P.Val) / length(treatedTT3HER2Basal$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3HER2Basal$adj.P.Val) / length(ntreatedTT3HER2Basal$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4HER2Basal$adj.P.Val) / length(treatedTT4HER2Basal$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4HER2Basal$adj.P.Val) / length(ntreatedTT4HER2Basal$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```












































