
```{r, include = FALSE}
library(ggplot2)
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools)
library(MASS)
library(splines)
library(tidyverse)
library(interactions)
library(readxl)
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(NOISeq)
```

reading in the expression count data and removing "EnsembleID" from first entry of list of samples:
```{r}
data <- read.delim("~/Documents/project/BRCA_Primary_Tumor_HTSeq_counts.txt.txt")
data <- data[c(1:60483),] #removing the last few genes from the dataframe which do not have an Ensembl ID assigned to them

matrix <- data
colnames(matrix) <- gsub('.', '-', colnames(matrix), fixed = TRUE)

sampleslist <- colnames(matrix)
sampleslist <- sampleslist[-1]
```

get TCGA breast cancer subtypes dataframe of the primary tumour expression data samples:
```{r }
CancerProject <- "TCGA-BRCA"
query <- GDCquery(project = CancerProject, #getinfo on the samples I have
                  data.category = "Transcriptome Profiling",
                  data.type = "Gene Expression Quantification", 
                  workflow.type = "HTSeq - Counts",
                  barcode = sampleslist)
samplesDown <- getResults(query,cols=c("cases"))
samplesDown1 <- getResults(query)
primarytumour <- TCGAquery_SampleTypes(barcode = samplesDown, typesample = "TP") #find out which are primary tumour
laterquery <- GDCquery(project = CancerProject,
                  data.category = "Transcriptome Profiling",
                  data.type = "Gene Expression Quantification", 
                  workflow.type = "HTSeq - Counts",
                  barcode = primarytumour) #get info on the primary tumour samples of my data only
samplesDownloaded <- getResults(laterquery,cols=c("cases"))
datasubtypes <- TCGA_MolecularSubtype(samplesDownloaded) #get subtypes of this data
datasubtypes <- datasubtypes$subtypes
datasubtypes_m <- datasubtypes[complete.cases(datasubtypes),] #remove NAs
```

get the matrix of expression data TCGA sample barcode clinical data:
```{r }
clindata <- GDCdownload(laterquery)
clinicaldata <- GDCprepare(laterquery)
colData(clinicaldata)
barcodesofEXPdata <- as.data.frame(colData(clinicaldata))
barcodesofEXPdata <- barcodesofEXPdata[which(barcodesofEXPdata$gender == "female"),] # filter out the samples that are not female 
```

Order the data and match up the dataframes:
```{r}
matrix_temp <- matrix[,colnames(matrix) %in% c("ENSEMBL_GeneID", barcodesofEXPdata$patient)]
matrix_temp <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", datasubtypes_m$patients)]# remove samples for which I don't have a subtype and which are not primary tumour
matrix_temp[,1] <- substr(matrix_temp[,1], 1, nchar("ENSG00000000003"))
rownames(matrix_temp) <- matrix_temp[,1]
matrix_temp <- matrix_temp[,-1]
#expression matrix for patients for which subtype data is KNOWN and duplicates are removed 
datasubtypes_mytemp <- datasubtypes_m[datasubtypes_m$patients %in% colnames(matrix_temp),]

#write.table(datasubtypes_my , file="~/Documents/project/datasubtypes_my.txt", sep="\t", quote=F)
matrix_temp <- matrix_temp[,order(datasubtypes_mytemp$patients)]
all.equal(as.character(colnames(matrix_temp)), datasubtypes_mytemp$patients)
dim(matrix_temp)
```

import tumour purity data:

```{r}
    barcodes <- as.character(datasubtypes_mytemp$samples)
    samples.id <- sapply(barcodes, function(x) paste(unlist(stringr::str_split(x, "-"))[1:4], collapse = "-"))
    df.barcodes_sampID <- data.frame(barcodes=barcodes, sampID=samples.id, row.names = 1:length(barcodes))
    filt.s <- c()
    for(s in samples.id){
        if(s %in% Tumor.purity$Sample.ID==FALSE)
            filt.s <- c(filt.s, s)
    }
    samples.filtered<-unlist(samples.id[samples.id %in% filt.s == FALSE])
    Tumor.purity.L <- Tumor.purity
    Tumor.purity.L <- Tumor.purity.L[which(Tumor.purity.L$Sample.ID %in% samples.filtered),]
    Tumor.purity.L$barcodes <- as.character(datasubtypes_mytemp$samples)
    Tumor.purity.L$ESTIMATE <- as.numeric(gsub(",", ".", Tumor.purity.L$ESTIMATE))
    Tumor.purity.L$ABSOLUTE <- as.numeric(gsub(",", ".", Tumor.purity.L$ABSOLUTE))
    Tumor.purity.L$LUMP <- as.numeric(gsub(",", ".", Tumor.purity.L$LUMP))
    Tumor.purity.L$IHC <- as.numeric(gsub(",", ".", Tumor.purity.L$IHC))
    Tumor.purity.L$CPE <- as.numeric(gsub(",", ".", Tumor.purity.L$CPE))
    rownames(Tumor.purity.L) <- substr(Tumor.purity.L$Sample.ID, 1, nchar("TCGA-3C-AAAU"))

save(Tumor.purity.L, file="~/Documents/project/Tumor.purity.L.Rdata")
```

Match up the subtype dataframes:
```{r}
rownames(datasubtypes_mytemp) <- datasubtypes_mytemp$patients
datasubtypes_my <- datasubtypes_mytemp
datasubtypes_my$subtype <- factor(datasubtypes_my$subtype)
all.equal(rownames(datasubtypes_my), rownames(Tumor.purity.L))
datasubtypes_my$CPE <- Tumor.purity.L$CPE
datasubtypes_my$IHC <- Tumor.purity.L$IHC
datasubtypes_my<-datasubtypes_my[complete.cases(datasubtypes_my$CPE),]

save(datasubtypes_my, file="~/Documents/project/datasubtypes_my.Rdata")

matrix_temp <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", datasubtypes_my$patients)]
save(matrix_temp, file="~/Documents/project/matrix_temp.Rdata")
lumAsub <- datasubtypes_my[which(datasubtypes_my$subtype == "BRCA.LumA"),]
lumBsub <- datasubtypes_my[which(datasubtypes_my$subtype == "BRCA.LumB"),]
Basalsub <- datasubtypes_my[which(datasubtypes_my$subtype == "BRCA.Basal"),]
Normalsub <- datasubtypes_my[which(datasubtypes_my$subtype == "BRCA.Normal"),]
HER2sub <- datasubtypes_my[which(datasubtypes_my$subtype == "BRCA.Her2"),]#

lumAlumBdata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumAsub$patients, lumBsub$patients)]
lumABasaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumAsub$patients, Basalsub$patients)]
lumAHER2data <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumAsub$patients, HER2sub$patients)]
lumANormaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumAsub$patients, Normalsub$patients)]
lumBBasaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumBsub$patients, Basalsub$patients)]
lumBNormaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumBsub$patients, Normalsub$patients)]
lumBHER2data <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", lumBsub$patients, HER2sub$patients)]
HER2Basaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", HER2sub$patients, Basalsub$patients)]
BasalNormaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", Basalsub$patients, Normalsub$patients)]
HER2Normaldata <- matrix_temp[,colnames(matrix_temp) %in% c("ENSEMBL_GeneID", HER2sub$patients, Normalsub$patients)]
save(lumAlumBdata, lumABasaldata, lumAHER2data, lumANormaldata, lumBBasaldata, lumBNormaldata, lumBHER2data, HER2Basaldata, BasalNormaldata, HER2Normaldata, file="subtypedata.Rdata")
dds1CPE <- DESeqDataSetFromMatrix(countData = as.matrix(matrix_temp), colData = datasubtypes_my, design = ~ CPE)
dds1CPE <- estimateSizeFactors(dds1CPE)
normalized_counts1CPE <- counts(dds1CPE, normalized=TRUE)
yCPE <- as.data.frame(log2(normalized_counts1CPE+1.0))
save(yCPE, datasubtypes_my, file="~/Documents/project/normalisedALLandDATASUBTYPES_MY.Rdata")
```


