

```{r, include = FALSE} 
library(ggplot2)#7 rows
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }
load(file="~/Documents/project/normaliseddataCPE.Rdata")
lumANormal <- datalumANormal
load(file="~/Documents/project/modelmatricesCPE.Rdata")
```



read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XlumANormal <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumANormal))
fitlumANormal <- lmFit(as.data.frame(lumANormal), XlumANormal)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XlumANormal))
#contr.matrix
#contrastsfitlumANormal <- contrasts.fit(fitlumANormal, contrasts=contr.matrix)
#contrastsfitlumANormal <- eBayes(contrastsfitlumANormal)
fitlumANormal <- eBayes(fitlumANormal)
TTlumANormal <- topTable(fitlumANormal, number = dim(lumANormal)[1])
pvalslumANormal <- fitlumANormal$p.value
fdradjusted_pvalslumANormal <- p.adjust(pvalslumANormal[,4], method = "fdr")
fdradjusted_pvalslumANormalsig <- fdradjusted_pvalslumANormal[which(fdradjusted_pvalslumANormal < 0.05)]
names(fdradjusted_pvalslumANormalsig)
top95lumANormal <- TTlumANormal[which(rownames(TTlumANormal) %in% names(fdradjusted_pvalslumANormalsig)),]
#plotMA(treatedTT4lumANormal , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTlumANormal$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitlumANormal, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumANormal)
treatedTT1lumANormal <- topTreat(fitlumANormal, coef=1, number=dim(lumANormal)[1], lfc=1)
treatedTT1lumANormalsig <- treatedTT1lumANormal[which(treatedTT1lumANormal$adj.P.Val < 0.05),]
treatedTT1lumANormalsig
treatedTT2lumANormal <- topTreat(fitlumANormal, coef=2, number=dim(lumANormal)[1], lfc=1)
treatedTT2lumANormalsig <- treatedTT2lumANormal[which(treatedTT2lumANormal$adj.P.Val < 0.05),]
treatedTT2lumANormalsig
treatedTT3lumANormal <- topTreat(fitlumANormal, coef=3, number=dim(lumANormal)[1], lfc=1)
treatedTT3lumANormalsig <- treatedTT3lumANormal[which(treatedTT3lumANormal$adj.P.Val < 0.05),]
treatedTT3lumANormalsig
treatedTT4lumANormal <- topTreat(fitlumANormal, coef=4, number=dim(lumANormal)[1], lfc=1)
treatedTT4lumANormalsig <- treatedTT4lumANormal[which(treatedTT4lumANormal$adj.P.Val < 0.05),]
treatedTT4lumANormalsig
#plot(density(treatedTT2lumANormalsig$logFC))
#ENSG00000122566
#ENSG00000122566

save(treatedTT4lumANormalsig, treatedTT2lumANormalsig, file="lumANormaloutputCPE.Rdata")
```






```{r}
plot(treatedTT4lumANormal$logFC, -log10(treatedTT4lumANormal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumANormalsig$logFC, -log10(treatedTT4lumANormalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumANormal <-model_lumANormal
nmodel_lumANormal$CPE <- (1- nmodel_lumANormal$CPE)
nXlumANormal <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumANormal))
nfitlumANormal <- lmFit(as.data.frame(lumANormal), nXlumANormal)

nfitlumANormal <- eBayes(nfitlumANormal)
nTTlumANormal<- topTable(nfitlumANormal, number = dim(lumANormal)[1])
npvalslumANormal <- nfitlumANormal$p.value
nfdradjusted_pvalslumANormal <- p.adjust(npvalslumANormal[,4], method = "fdr")
nfdradjusted_pvalslumANormalsig <- nfdradjusted_pvalslumANormal[which(nfdradjusted_pvalslumANormal < 0.05)]
names(nfdradjusted_pvalslumANormalsig)
ntop95lumANormal <- nTTlumANormal[which(rownames(nTTlumANormal) %in% names(nfdradjusted_pvalslumANormalsig)),]

ntreatedTT1lumANormal <- topTreat(nfitlumANormal, coef=1, number=dim(lumANormal)[1], lfc=1)
ntreatedTT1lumANormalsig <- ntreatedTT1lumANormal[which(ntreatedTT1lumANormal$adj.P.Val < 0.05),]
ntreatedTT1lumANormalsig
ntreatedTT2lumANormal <- topTreat(nfitlumANormal, coef=2, number=dim(lumANormal)[1], lfc=1)
ntreatedTT2lumANormalsig <- ntreatedTT2lumANormal[which(ntreatedTT2lumANormal$adj.P.Val < 0.05),]
ntreatedTT2lumANormalsig
ntreatedTT3lumANormal <- topTreat(nfitlumANormal, coef=3, number=dim(lumANormal)[1], lfc=1)
ntreatedTT3lumANormalsig <- ntreatedTT3lumANormal[which(ntreatedTT3lumANormal$adj.P.Val < 0.05),]
ntreatedTT3lumANormalsig
ntreatedTT4lumANormal <- topTreat(nfitlumANormal, coef=4, number=dim(lumANormal)[1], lfc=1)
ntreatedTT4lumANormalsig <- ntreatedTT4lumANormal[which(ntreatedTT4lumANormal$adj.P.Val < 0.05),]
ntreatedTT4lumANormalsig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumANormal <- as.data.frame(model_lumANormal)
nCPEmodel_lumANormal$CPE <- (nCPEmodel_lumANormal$CPE)
nthelm2lumANormal <- lm(lumANormal[17876,] ~ subtype * CPE, data=nCPEmodel_lumANormal)

plot_model(nthelm2lumANormal, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumANormal) == "ENSG00000214688") # this is a gene responsible for ER expression
```

```{r}
normallumANormal <- treatedTT2lumANormalsig[which(rownames(treatedTT2lumANormalsig) %in% rownames(treatedTT4lumANormalsig)),]
normallumANormal <- normallumANormal[which(!rownames(normallumANormal) %in% rownames(ntreatedTT2lumANormalsig)),]

cancerlumANormal <- ntreatedTT2lumANormalsig[which(rownames(ntreatedTT2lumANormalsig) %in% rownames(ntreatedTT4lumANormalsig)),]
cancerlumANormal <- cancerlumANormal[which(!rownames(cancerlumANormal) %in% rownames(treatedTT2lumANormalsig)),]
cancerlumANormal
normallumANormal 


save(cancerlumANormal, normallumANormal, file="lumANormal_C_N_CPE.Rdata")
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumANormalsig$status <- ifelse(rownames(treatedTT4lumANormalsig) %in% rownames(cancerlumANormal), "Within Cancer Cells", ifelse(rownames(treatedTT4lumANormalsig) %in% rownames(normallumANormal), "Within Normal Cells", "Both"))
treatedTT4lumANormalsig$colour <- ifelse(rownames(treatedTT4lumANormalsig) %in% rownames(cancerlumANormal), cb3[2], ifelse(rownames(treatedTT4lumANormalsig) %in% rownames(normallumANormal), cb3[3], cb3[1]))
geneslumANormal <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumANormalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumANormal <- geneslumANormal[!duplicated(geneslumANormal$GENEID),]
geneslumANormal$logFC <- treatedTT4lumANormalsig[which(rownames(treatedTT4lumANormalsig) %in% geneslumANormal$GENEID),][geneslumANormal$GENEID,]$logFC
geneslumANormal$status <- treatedTT4lumANormalsig[which(rownames(treatedTT4lumANormalsig) %in% geneslumANormal$GENEID),][geneslumANormal$GENEID,]$status
lumANormalsigs <- ggplot(geneslumANormal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Normal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumANormalsigsCPE.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumANormal <-geneslumANormal[which(geneslumANormal$status != "Both"),]
cancernormalonlylumANormalgraph <- ggplot(cancernormalonlylumANormal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Normal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumANormalCPE.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XlumANormals <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumANormal))
fitlumANormals <- lmFit(as.data.frame(lumANormal), XlumANormals)
fitlumANormals <- eBayes(fitlumANormals)
TTlumANormals <- topTable(fitlumANormals, number = dim(lumANormal)[1])
pvalslumANormals <- fitlumANormals$p.value
fdradjusted_pvalslumANormalINTERs <- p.adjust(pvalslumANormals[,1], method = "fdr")
fdradjusted_pvalslumANormalsigINTERs <- fdradjusted_pvalslumANormalINTERs[which(fdradjusted_pvalslumANormalINTERs < 0.05)]
length(names(fdradjusted_pvalslumANormalsigINTERs))
fdradjusted_pvalslumANormalSUBs <- p.adjust(pvalslumANormals[,2], method = "fdr")
fdradjusted_pvalslumANormalsigSUBs <- fdradjusted_pvalslumANormalSUBs[which(fdradjusted_pvalslumANormalSUBs < 0.05)]
length(names(fdradjusted_pvalslumANormalsigSUBs))
fdradjusted_pvalslumANormalCPEs <- p.adjust(pvalslumANormals[,3], method = "fdr")
fdradjusted_pvalslumANormalsigCPEs <- fdradjusted_pvalslumANormalCPEs[which(fdradjusted_pvalslumANormalCPEs < 0.05)]
length(names(fdradjusted_pvalslumANormalsigCPEs))
plotSA(fitlumANormals, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumANormalSUBs)

treatedTT1lumANormals <- topTreat(fitlumANormals, coef=1, number=dim(lumANormal)[1])
treatedTT1lumANormalsigs <- treatedTT1lumANormals[which(treatedTT1lumANormals$adj.P.Val < 0.05),]
treatedTT1lumANormalsigs
treatedTT2lumANormals <- topTreat(fitlumANormals, coef=2, number=dim(lumANormal)[1])
treatedTT2lumANormalsigs <- treatedTT2lumANormals[which(treatedTT2lumANormals$adj.P.Val < 0.05),]
treatedTT2lumANormalsigs
treatedTT3lumANormals <- topTreat(fitlumANormals, coef=3, number=dim(lumANormal)[1])
treatedTT3lumANormalsigs <- treatedTT3lumANormals[which(treatedTT3lumANormals$adj.P.Val < 0.05),]
treatedTT3lumANormalsigs


treatedTT2lumANormalsigs[which(rownames(treatedTT2lumANormalsigs) %in% rownames(treatedTT4lumANormalsig)),]
treatedTT2lumANormalsigs[which(rownames(treatedTT2lumANormalsigs) %in% rownames(cancerlumANormal)),]
treatedTT2lumANormalsigs[which(rownames(treatedTT2lumANormalsigs) %in% rownames(normallumANormal)),]
```

```{r}
plot(treatedTT4lumANormal$logFC, -log10(treatedTT4lumANormal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumANormalsig$logFC, -log10(treatedTT4lumANormalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumANormals <- model_lumANormal
nmodel_lumANormals$CPE <- (1- nmodel_lumANormals$CPE)
nXlumANormals <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumANormals))
nfitlumANormals <- lmFit(as.data.frame(lumANormal), nXlumANormals)
nfitlumANormals <- eBayes(nfitlumANormals)
nTTlumANormals<- topTable(nfitlumANormals, number = dim(lumANormal)[1])
npvalslumANormals <- nfitlumANormals$p.value
ntreatedTT1lumANormals <- topTreat(nfitlumANormals, coef=1, number=dim(lumANormal)[1])
ntreatedTT1lumANormalsigs <- ntreatedTT1lumANormals[which(ntreatedTT1lumANormals$adj.P.Val < 0.05),]
ntreatedTT1lumANormalsigs
ntreatedTT2lumANormals <- topTreat(nfitlumANormals, coef=2, number=dim(lumANormal)[1])
ntreatedTT2lumANormalsigs <- ntreatedTT2lumANormals[which(ntreatedTT2lumANormals$adj.P.Val < 0.05),]
ntreatedTT2lumANormalsigs
#ntreatedTT2lumANormalsigs[which(rownames(ntreatedTT2lumANormalsigs) %in% rownames(normallumANormal)),]
ntreatedTT3lumANormals <- topTreat(nfitlumANormals, coef=3, number=dim(lumANormal)[1])
ntreatedTT3lumANormalsigs <- ntreatedTT3lumANormals[which(ntreatedTT3lumANormals$adj.P.Val < 0.05),]
ntreatedTT3lumANormalsigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumANormals <- as.data.frame(nmodel_lumANormals)
nCPEmodel_lumANormals$CPE <- (1- nCPEmodel_lumANormals$CPE)
nthelm2lumANormals <- lm(lumANormal[5015,] ~ subtype + CPE, data=nCPEmodel_lumANormals)

plot_model(nthelm2lumANormals, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumANormal) == "ENSG00000122566")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2lumANormals$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2lumANormals$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal A", "Normal-like")
CPEmodel_lumANormal <- as.data.frame(model_lumANormal)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumANormalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumANormal <- lumANormal[which(rownames(lumANormal) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumANormal[top10Genes$GENEID,]
top10GeneslumANormal <- top10GeneslumANormal[order(rownames(top10GeneslumANormal)),]
dim(top10GeneslumANormal)
all.equal(rownames(top10GeneslumANormal), top10Genes$GENEID)
rownames(top10GeneslumANormal) <- top10Genes$SYMBOL
top10lmlumANormal <-  numeric(length = nrow(top10GeneslumANormal))
plotlumANormalCPE1 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE1, file=paste0("~/Documents/project/plots/lumANormalCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumANormalCPE2 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE2, file=paste0("~/Documents/project/plots/lumANormalCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumANormalCPE3 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE3, file=paste0("~/Documents/project/plots/lumANormalCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumANormalCPE4 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE4, file=paste0("~/Documents/project/plots/lumANormalCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumANormalCPE5 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE5, file=paste0("~/Documents/project/plots/lumANormalCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumANormalCPE6 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE6, file=paste0("~/Documents/project/plots/lumANormalCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumANormalCPE7 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE7, file=paste0("~/Documents/project/plots/lumANormalCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumANormalCPE8 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE8, file=paste0("~/Documents/project/plots/lumANormalCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumANormalCPE9 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumANormal)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE9, file=paste0("~/Documents/project/plots/lumANormalCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumANormalCPE10 <- ggplot(CPEmodel_lumANormal ,aes(y=top10GeneslumANormal[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumANormal)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumANormalCPE10, file=paste0("~/Documents/project/plots/lumANormalCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumANormal <-cowplot::plot_grid(plotlumANormalCPE1, plotlumANormalCPE2 + theme(legend.position="none"), plotlumANormalCPE3 + theme(legend.position="none"),plotlumANormalCPE4 + theme(legend.position="none"), plotlumANormalCPE5 + theme(legend.position="none"), plotlumANormalCPE6 + theme(legend.position="none"),plotlumANormalCPE7 + theme(legend.position="none"), plotlumANormalCPE8 + theme(legend.position="none"), plotlumANormalCPE9 + theme(legend.position="none"),plotlumANormalCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumANormal, file=paste0("~/Documents/project/plots/mergedCPElumANormal.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XlumANormalss <- model.matrix(~ subtype, data=as.data.frame(model_lumANormal))
fitlumANormalss <- lmFit(as.data.frame(lumANormal), XlumANormalss)

fitlumANormalss <- eBayes(fitlumANormalss)
TTlumANormalss <- topTable(fitlumANormalss, number = dim(lumANormal)[1])
pvalslumANormalss <- fitlumANormalss$p.value
fdradjusted_pvalslumANormalINTERss <- p.adjust(pvalslumANormalss[,1], method = "fdr")
fdradjusted_pvalslumANormalsigINTERss <- fdradjusted_pvalslumANormalINTERss[which(fdradjusted_pvalslumANormalINTERss < 0.05)]
length(names(fdradjusted_pvalslumANormalsigINTERss))
plotSA(fitlumANormalss, main="Final model: Mean-variance trend")

treatedTT1lumANormalss <- topTreat(fitlumANormalss, coef=1, number=dim(lumANormal)[1], lfc=1)
treatedTT1lumANormalsigss <- treatedTT1lumANormalss[which(treatedTT1lumANormalss$adj.P.Val < 0.05),]
treatedTT1lumANormalsigss
```

```{r}
plot(treatedTT1lumANormalss$logFC, -log10(treatedTT1lumANormalss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1lumANormalsigss$logFC, -log10(treatedTT1lumANormalsigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4lumANormal$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4lumANormal$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4lumANormal$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumANormal$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumANormal$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumANormal$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumANormal$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumANormal$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumANormal$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumANormal$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumANormals$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumANormals$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumANormals$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumANormals$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumANormals$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumANormals$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumANormalsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1lumANormals$adj.P.Val) / length(ntreatedTT1lumANormals$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2lumANormals$adj.P.Val) / length(ntreatedTT2lumANormals$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3lumANormals$adj.P.Val) / length(ntreatedTT3lumANormals$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1lumANormalss$adj.P.Val) / length(treatedTT1lumANormalss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumANormal$adj.P.Val) / length(ntreatedTT4lumANormal$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2lumANormal$adj.P.Val) / length(ntreatedTT2lumANormal$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3lumANormal$adj.P.Val) / length(ntreatedTT3lumANormal$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1lumANormal$adj.P.Val) / length(ntreatedTT1lumANormal$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumANormal$adj.P.Val) / length(ntreatedTT1lumANormal$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumANormal$adj.P.Val) / length(ntreatedTT4lumANormal$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumANormal$adj.P.Val) / length(ntreatedTT2lumANormal$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumANormal$adj.P.Val) / length(ntreatedTT3lumANormal$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumANormal$adj.P.Val) / length(treatedTT1lumANormal$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumANormal$adj.P.Val) / length(treatedTT3lumANormal$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumANormal$adj.P.Val) / length(treatedTT4lumANormal$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumANormal$adj.P.Val) / length(treatedTT2lumANormal$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2lumANormal$adj.P.Val) / length(treatedTT2lumANormal$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumANormal$adj.P.Val) / length(ntreatedTT2lumANormal$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumANormal$adj.P.Val) / length(treatedTT3lumANormal$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumANormal$adj.P.Val) / length(ntreatedTT3lumANormal$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumANormal$adj.P.Val) / length(treatedTT4lumANormal$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumANormal$adj.P.Val) / length(ntreatedTT4lumANormal$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```




















