```{r, include = FALSE} 
library(ggplot2)
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools)
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(GSVAdata)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```


```{r }

load(file="~/Documents/project/normaliseddataCPE.Rdata")
lumABasal <- datalumABasal
load(file="~/Documents/project/modelmatricesCPE.Rdata")
```

the MAIN model:
```{r}
XlumABasal <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumABasal))
fitlumABasal <- lmFit(as.data.frame(lumABasal), XlumABasal)

fitlumABasal <- eBayes(fitlumABasal)
TTlumABasal <- topTable(fitlumABasal, number = dim(lumABasal)[1])
pvalslumABasal <- fitlumABasal$p.value
fdradjusted_pvalslumABasal <- p.adjust(pvalslumABasal[,4], method = "fdr")
fdradjusted_pvalslumABasalsig <- fdradjusted_pvalslumABasal[which(fdradjusted_pvalslumABasal < 0.05)]
names(fdradjusted_pvalslumABasalsig)
top95lumABasal <- TTlumABasal[which(rownames(TTlumABasal) %in% names(fdradjusted_pvalslumABasalsig)),]
#plotMA(treatedTT4lumABasal , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTlumABasal$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitlumABasal, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumABasal)
treatedTT1lumABasal <- topTreat(fitlumABasal, coef=1, number=dim(lumABasal)[1], lfc=1)
treatedTT1lumABasalsig <- treatedTT1lumABasal[which(treatedTT1lumABasal$adj.P.Val < 0.05),]
treatedTT1lumABasalsig
treatedTT2lumABasal <- topTreat(fitlumABasal, coef=2, number=dim(lumABasal)[1], lfc=1)
treatedTT2lumABasalsig <- treatedTT2lumABasal[which(treatedTT2lumABasal$adj.P.Val < 0.05),]
treatedTT2lumABasalsig
treatedTT3lumABasal <- topTreat(fitlumABasal, coef=3, number=dim(lumABasal)[1], lfc=1)
treatedTT3lumABasalsig <- treatedTT3lumABasal[which(treatedTT3lumABasal$adj.P.Val < 0.05),]
treatedTT3lumABasalsig
treatedTT4lumABasal <- topTreat(fitlumABasal, coef=4, number=dim(lumABasal)[1], lfc=1)
treatedTT4lumABasalsig <- treatedTT4lumABasal[which(treatedTT4lumABasal$adj.P.Val < 0.05),]
treatedTT4lumABasalsig

save(treatedTT4lumABasalsig, treatedTT2lumABasalsig, file="lumABasaloutputCPE.Rdata")
```






```{r}
plot(treatedTT4lumABasal$logFC, -log10(treatedTT4lumABasal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumABasalsig$logFC, -log10(treatedTT4lumABasalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear model:

```{r}
nmodel_lumABasal <-model_lumABasal
nmodel_lumABasal$CPE <- (1- nmodel_lumABasal$CPE)
nXlumABasal <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumABasal))
nfitlumABasal <- lmFit(as.data.frame(lumABasal), nXlumABasal)

nfitlumABasal <- eBayes(nfitlumABasal)
nTTlumABasal<- topTable(nfitlumABasal, number = dim(lumABasal)[1])
npvalslumABasal <- nfitlumABasal$p.value
nfdradjusted_pvalslumABasal <- p.adjust(npvalslumABasal[,4], method = "fdr")
nfdradjusted_pvalslumABasalsig <- nfdradjusted_pvalslumABasal[which(nfdradjusted_pvalslumABasal < 0.05)]
names(nfdradjusted_pvalslumABasalsig)
ntop95lumABasal <- nTTlumABasal[which(rownames(nTTlumABasal) %in% names(nfdradjusted_pvalslumABasalsig)),]

ntreatedTT1lumABasal <- topTreat(nfitlumABasal, coef=1, number=dim(lumABasal)[1], lfc=1)
ntreatedTT1lumABasalsig <- ntreatedTT1lumABasal[which(ntreatedTT1lumABasal$adj.P.Val < 0.05),]
ntreatedTT1lumABasalsig
ntreatedTT2lumABasal <- topTreat(nfitlumABasal, coef=2, number=dim(lumABasal)[1], lfc=1)
ntreatedTT2lumABasalsig <- ntreatedTT2lumABasal[which(ntreatedTT2lumABasal$adj.P.Val < 0.05),]
ntreatedTT2lumABasalsig
ntreatedTT3lumABasal <- topTreat(nfitlumABasal, coef=3, number=dim(lumABasal)[1], lfc=1)
ntreatedTT3lumABasalsig <- ntreatedTT3lumABasal[which(ntreatedTT3lumABasal$adj.P.Val < 0.05),]
ntreatedTT3lumABasalsig
ntreatedTT4lumABasal <- topTreat(nfitlumABasal, coef=4, number=dim(lumABasal)[1], lfc=1)
ntreatedTT4lumABasalsig <- ntreatedTT4lumABasal[which(ntreatedTT4lumABasal$adj.P.Val < 0.05),]
ntreatedTT4lumABasalsig
```

single linear model with no interaction for 1-tumour purity:
```{r}
nCPEmodel_lumABasal <- as.data.frame(model_lumABasal)
nCPEmodel_lumABasal$CPE <- (nCPEmodel_lumABasal$CPE)
nthelm2lumABasal <- lm(lumABasal[17876,] ~ subtype * CPE, data=nCPEmodel_lumABasal)

plot_model(nthelm2lumABasal, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumABasal) == "ENSG00000214688") # this is a gene responsible for ER expression
```
finding the significantly differentally expressed genes specifically within the normal and specifically within the cancer cells respectively using the main model and CPE as tumour purity:
```{r}
normallumABasal <- treatedTT2lumABasalsig[which(rownames(treatedTT2lumABasalsig) %in% rownames(treatedTT4lumABasalsig)),]
normallumABasal <- normallumABasal[which(!rownames(normallumABasal) %in% rownames(ntreatedTT2lumABasalsig)),]

cancerlumABasal <- ntreatedTT2lumABasalsig[which(rownames(ntreatedTT2lumABasalsig) %in% rownames(ntreatedTT4lumABasalsig)),]
cancerlumABasal <- cancerlumABasal[which(!rownames(cancerlumABasal) %in% rownames(treatedTT2lumABasalsig)),]
cancerlumABasal
normallumABasal 


save(cancerlumABasal, normallumABasal, file="lumABasal_C_N_CPE.Rdata")
```


#plot the signifcantly differentially expressed genes specifically within the cancer, normal and both and then specifically within the normal and cancer only:

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumABasalsig$status <- ifelse(rownames(treatedTT4lumABasalsig) %in% rownames(cancerlumABasal), "Within Cancer Cells", ifelse(rownames(treatedTT4lumABasalsig) %in% rownames(normallumABasal), "Within Normal Cells", "Both"))
treatedTT4lumABasalsig$colour <- ifelse(rownames(treatedTT4lumABasalsig) %in% rownames(cancerlumABasal), cb3[2], ifelse(rownames(treatedTT4lumABasalsig) %in% rownames(normallumABasal), cb3[3], cb3[1]))
geneslumABasal <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumABasalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumABasal <- geneslumABasal[!duplicated(geneslumABasal$GENEID),]
geneslumABasal <- geneslumABasal[!duplicated(geneslumABasal$SYMBOL),]

geneslumABasal$logFC <- treatedTT4lumABasalsig[which(rownames(treatedTT4lumABasalsig) %in% geneslumABasal$GENEID),][geneslumABasal$GENEID,]$logFC
geneslumABasal$status <- treatedTT4lumABasalsig[which(rownames(treatedTT4lumABasalsig) %in% geneslumABasal$GENEID),][geneslumABasal$GENEID,]$status
lumABasalsigs <- ggplot(geneslumABasal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumABasalsigsCPE.png")


mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumABasal <-geneslumABasal[which(geneslumABasal$status != "Both"),]
cancernormalonlylumABasalgraph <- ggplot(cancernormalonlylumABasal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumABasalCPE.png")
```

Simple linear model NO interaciton:

```{r}
XlumABasals <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumABasal))
fitlumABasals <- lmFit(as.data.frame(lumABasal), XlumABasals)
fitlumABasals <- eBayes(fitlumABasals)
TTlumABasals <- topTable(fitlumABasals, number = dim(lumABasal)[1])
pvalslumABasals <- fitlumABasals$p.value
fdradjusted_pvalslumABasalINTERs <- p.adjust(pvalslumABasals[,1], method = "fdr")
fdradjusted_pvalslumABasalsigINTERs <- fdradjusted_pvalslumABasalINTERs[which(fdradjusted_pvalslumABasalINTERs < 0.05)]
length(names(fdradjusted_pvalslumABasalsigINTERs))
fdradjusted_pvalslumABasalSUBs <- p.adjust(pvalslumABasals[,2], method = "fdr")
fdradjusted_pvalslumABasalsigSUBs <- fdradjusted_pvalslumABasalSUBs[which(fdradjusted_pvalslumABasalSUBs < 0.05)]
length(names(fdradjusted_pvalslumABasalsigSUBs))
fdradjusted_pvalslumABasalCPEs <- p.adjust(pvalslumABasals[,3], method = "fdr")
fdradjusted_pvalslumABasalsigCPEs <- fdradjusted_pvalslumABasalCPEs[which(fdradjusted_pvalslumABasalCPEs < 0.05)]
length(names(fdradjusted_pvalslumABasalsigCPEs))
plotSA(fitlumABasals, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumABasalSUBs)

treatedTT1lumABasals <- topTreat(fitlumABasals, coef=1, number=dim(lumABasal)[1], lfc=1)
treatedTT1lumABasalsigs <- treatedTT1lumABasals[which(treatedTT1lumABasals$adj.P.Val < 0.05),]
treatedTT1lumABasalsigs
treatedTT2lumABasals <- topTreat(fitlumABasals, coef=2, number=dim(lumABasal)[1], lfc=1)
treatedTT2lumABasalsigs <- treatedTT2lumABasals[which(treatedTT2lumABasals$adj.P.Val < 0.05),]
treatedTT2lumABasalsigs
treatedTT3lumABasals <- topTreat(fitlumABasals, coef=3, number=dim(lumABasal)[1], lfc=1)
treatedTT3lumABasalsigs <- treatedTT3lumABasals[which(treatedTT3lumABasals$adj.P.Val < 0.05),]
treatedTT3lumABasalsigs


treatedTT2lumABasalsigs[which(rownames(treatedTT2lumABasalsigs) %in% rownames(treatedTT4lumABasalsig)),]
treatedTT2lumABasalsigs[which(rownames(treatedTT2lumABasalsigs) %in% rownames(cancerlumABasal)),]
treatedTT2lumABasalsigs[which(rownames(treatedTT2lumABasalsigs) %in% rownames(normallumABasal)),]
```

```{r}
plot(treatedTT4lumABasal$logFC, -log10(treatedTT4lumABasal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumABasalsig$logFC, -log10(treatedTT4lumABasalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity simple linear model:

```{r}
nmodel_lumABasals <- model_lumABasal
nmodel_lumABasals$CPE <- (1- nmodel_lumABasals$CPE)
nXlumABasals <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumABasals))
nfitlumABasals <- lmFit(as.data.frame(lumABasal), nXlumABasals)
nfitlumABasals <- eBayes(nfitlumABasals)
nTTlumABasals<- topTable(nfitlumABasals, number = dim(lumABasal)[1])
npvalslumABasals <- nfitlumABasals$p.value
ntreatedTT1lumABasals <- topTreat(nfitlumABasals, coef=1, number=dim(lumABasal)[1], lfc=1)
ntreatedTT1lumABasalsigs <- ntreatedTT1lumABasals[which(ntreatedTT1lumABasals$adj.P.Val < 0.05),]
ntreatedTT1lumABasalsigs
ntreatedTT2lumABasals <- topTreat(nfitlumABasals, coef=2, number=dim(lumABasal)[1], lfc=1)
ntreatedTT2lumABasalsigs <- ntreatedTT2lumABasals[which(ntreatedTT2lumABasals$adj.P.Val < 0.05),]
ntreatedTT2lumABasalsigs
#ntreatedTT2lumABasalsigs[which(rownames(ntreatedTT2lumABasalsigs) %in% rownames(normallumABasal)),]
ntreatedTT3lumABasals <- topTreat(nfitlumABasals, coef=3, number=dim(lumABasal)[1], lfc=1)
ntreatedTT3lumABasalsigs <- ntreatedTT3lumABasals[which(ntreatedTT3lumABasals$adj.P.Val < 0.05),]
ntreatedTT3lumABasalsigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumABasals <- as.data.frame(nmodel_lumABasals)
nCPEmodel_lumABasals$CPE <- (1- nCPEmodel_lumABasals$CPE)
nthelm2lumABasals <- lm(lumABasal[5015,] ~ subtype + CPE, data=nCPEmodel_lumABasals)

plot_model(nthelm2lumABasals, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumABasal) == "ENSG00000122566")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2lumABasals$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2lumABasals$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for the top 10 expressed genes in the data:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal A", "Basal-like")
CPEmodel_lumABasal <- as.data.frame(model_lumABasal)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumABasalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumABasal <- lumABasal[which(rownames(lumABasal) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumABasal[top10Genes$GENEID,]
top10GeneslumABasal <- top10GeneslumABasal[order(rownames(top10GeneslumABasal)),]
dim(top10GeneslumABasal)
all.equal(rownames(top10GeneslumABasal), top10Genes$GENEID)
rownames(top10GeneslumABasal) <- top10Genes$SYMBOL
top10lmlumABasal <-  numeric(length = nrow(top10GeneslumABasal))
plotlumABasalCPE1 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE1, file=paste0("~/Documents/project/plots/lumABasalCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumABasalCPE2 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE2, file=paste0("~/Documents/project/plots/lumABasalCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumABasalCPE3 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE3, file=paste0("~/Documents/project/plots/lumABasalCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumABasalCPE4 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE4, file=paste0("~/Documents/project/plots/lumABasalCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumABasalCPE5 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE5, file=paste0("~/Documents/project/plots/lumABasalCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumABasalCPE6 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE6, file=paste0("~/Documents/project/plots/lumABasalCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumABasalCPE7 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE7, file=paste0("~/Documents/project/plots/lumABasalCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumABasalCPE8 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE8, file=paste0("~/Documents/project/plots/lumABasalCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumABasalCPE9 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumABasal)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE9, file=paste0("~/Documents/project/plots/lumABasalCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumABasalCPE10 <- ggplot(CPEmodel_lumABasal ,aes(y=top10GeneslumABasal[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumABasal)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumABasalCPE10, file=paste0("~/Documents/project/plots/lumABasalCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumABasal <-cowplot::plot_grid(plotlumABasalCPE1, plotlumABasalCPE2 + theme(legend.position="none"), plotlumABasalCPE3 + theme(legend.position="none"),plotlumABasalCPE4 + theme(legend.position="none"), plotlumABasalCPE5 + theme(legend.position="none"), plotlumABasalCPE6 + theme(legend.position="none"),plotlumABasalCPE7 + theme(legend.position="none"), plotlumABasalCPE8 + theme(legend.position="none"), plotlumABasalCPE9 + theme(legend.position="none"),plotlumABasalCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumABasal, file=paste0("~/Documents/project/plots/mergedCPElumABasal.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

```{r}
XlumABasalss <- model.matrix(~ subtype, data=as.data.frame(model_lumABasal))
fitlumABasalss <- lmFit(as.data.frame(lumABasal), XlumABasalss)

fitlumABasalss <- eBayes(fitlumABasalss)
TTlumABasalss <- topTable(fitlumABasalss, number = dim(lumABasal)[1])
pvalslumABasalss <- fitlumABasalss$p.value
fdradjusted_pvalslumABasalINTERss <- p.adjust(pvalslumABasalss[,1], method = "fdr")
fdradjusted_pvalslumABasalsigINTERss <- fdradjusted_pvalslumABasalINTERss[which(fdradjusted_pvalslumABasalINTERss < 0.05)]
length(names(fdradjusted_pvalslumABasalsigINTERss))
plotSA(fitlumABasalss, main="Final model: Mean-variance trend")

treatedTT1lumABasalss <- topTreat(fitlumABasalss, coef=1, number=dim(lumABasal)[1], lfc=1)
treatedTT1lumABasalsigss <- treatedTT1lumABasalss[which(treatedTT1lumABasalss$adj.P.Val < 0.05),]
treatedTT1lumABasalsigss
```
plot a volcano plot of the significant genes:
```{r}
plot(treatedTT1lumABasalss$logFC, -log10(treatedTT1lumABasalss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1lumABasalsigss$logFC, -log10(treatedTT1lumABasalsigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots of adj p values:
```{r}

nconvSort4 <- sort(-log10(ntreatedTT4lumABasal$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumABasal$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumABasal$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumABasal$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumABasal$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumABasal$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumABasal$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumABasal$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumABasals$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumABasals$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumABasals$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumABasals$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumABasals$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumABasals$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumABasalsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1lumABasals$adj.P.Val) / length(ntreatedTT1lumABasals$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2lumABasals$adj.P.Val) / length(ntreatedTT2lumABasals$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3lumABasals$adj.P.Val) / length(ntreatedTT3lumABasals$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1lumABasalss$adj.P.Val) / length(treatedTT1lumABasalss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumABasal$adj.P.Val) / length(ntreatedTT4lumABasal$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2lumABasal$adj.P.Val) / length(ntreatedTT2lumABasal$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3lumABasal$adj.P.Val) / length(ntreatedTT3lumABasal$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1lumABasal$adj.P.Val) / length(ntreatedTT1lumABasal$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumABasal$adj.P.Val) / length(ntreatedTT1lumABasal$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumABasal$adj.P.Val) / length(ntreatedTT4lumABasal$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumABasal$adj.P.Val) / length(ntreatedTT2lumABasal$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumABasal$adj.P.Val) / length(ntreatedTT3lumABasal$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumABasal$adj.P.Val) / length(treatedTT1lumABasal$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumABasal$adj.P.Val) / length(treatedTT3lumABasal$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumABasal$adj.P.Val) / length(treatedTT4lumABasal$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumABasal$adj.P.Val) / length(treatedTT2lumABasal$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2lumABasal$adj.P.Val) / length(treatedTT2lumABasal$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumABasal$adj.P.Val) / length(ntreatedTT2lumABasal$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumABasal$adj.P.Val) / length(treatedTT3lumABasal$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumABasal$adj.P.Val) / length(ntreatedTT3lumABasal$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumABasal$adj.P.Val) / length(treatedTT4lumABasal$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumABasal$adj.P.Val) / length(ntreatedTT4lumABasal$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```


























