---
title: "LumAvsHER2"
output: html_document
---


```{r, include = FALSE} 
library(ggplot2)#7 rows
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }
load(file="~/Documents/project/normaliseddataCPE.Rdata")
lumAHER2 <- datalumAHER2
load(file="~/Documents/project/modelmatricesCPE.Rdata")
```




read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XlumAHER2 <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumAHER2))
fitlumAHER2 <- lmFit(as.data.frame(lumAHER2), XlumAHER2)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XlumAHER2))
#contr.matrix
#contrastsfitlumAHER2 <- contrasts.fit(fitlumAHER2, contrasts=contr.matrix)
#contrastsfitlumAHER2 <- eBayes(contrastsfitlumAHER2)
fitlumAHER2 <- eBayes(fitlumAHER2)
TTlumAHER2 <- topTable(fitlumAHER2, number = dim(lumAHER2)[1])
pvalslumAHER2 <- fitlumAHER2$p.value
fdradjusted_pvalslumAHER2 <- p.adjust(pvalslumAHER2[,4], method = "fdr")
fdradjusted_pvalslumAHER2sig <- fdradjusted_pvalslumAHER2[which(fdradjusted_pvalslumAHER2 < 0.05)]
names(fdradjusted_pvalslumAHER2sig)
top95lumAHER2 <- TTlumAHER2[which(rownames(TTlumAHER2) %in% names(fdradjusted_pvalslumAHER2sig)),]
plotSA(fitlumAHER2, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumAHER2)
treatedTT1lumAHER2 <- topTreat(fitlumAHER2, coef=1, number=dim(lumAHER2)[1], lfc=1)
treatedTT1lumAHER2sig <- treatedTT1lumAHER2[which(treatedTT1lumAHER2$adj.P.Val < 0.05),]
treatedTT1lumAHER2sig
treatedTT2lumAHER2 <- topTreat(fitlumAHER2, coef=2, number=dim(lumAHER2)[1], lfc=1)
treatedTT2lumAHER2sig <- treatedTT2lumAHER2[which(treatedTT2lumAHER2$adj.P.Val < 0.05),]
treatedTT2lumAHER2sig
treatedTT3lumAHER2 <- topTreat(fitlumAHER2, coef=3, number=dim(lumAHER2)[1], lfc=1)
treatedTT3lumAHER2sig <- treatedTT3lumAHER2[which(treatedTT3lumAHER2$adj.P.Val < 0.05),]
treatedTT3lumAHER2sig
treatedTT4lumAHER2 <- topTreat(fitlumAHER2, coef=4, number=dim(lumAHER2)[1], lfc=1)
treatedTT4lumAHER2sig <- treatedTT4lumAHER2[which(treatedTT4lumAHER2$adj.P.Val < 0.05),]
treatedTT4lumAHER2sig

save(treatedTT4lumAHER2sig, treatedTT2lumAHER2sig, file="lumAHER2outputCPE.Rdata")
```






```{r}
plot(treatedTT4lumAHER2$logFC, -log10(treatedTT4lumAHER2$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumAHER2sig$logFC, -log10(treatedTT4lumAHER2sig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumAHER2 <-model_lumAHER2
nmodel_lumAHER2$CPE <- (1- nmodel_lumAHER2$CPE)
nXlumAHER2 <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumAHER2))
nfitlumAHER2 <- lmFit(as.data.frame(lumAHER2), nXlumAHER2)

nfitlumAHER2 <- eBayes(nfitlumAHER2)
nTTlumAHER2<- topTable(nfitlumAHER2, number = dim(lumAHER2)[1])
npvalslumAHER2 <- nfitlumAHER2$p.value
nfdradjusted_pvalslumAHER2 <- p.adjust(npvalslumAHER2[,4], method = "fdr")
nfdradjusted_pvalslumAHER2sig <- nfdradjusted_pvalslumAHER2[which(nfdradjusted_pvalslumAHER2 < 0.05)]
names(nfdradjusted_pvalslumAHER2sig)
ntop95lumAHER2 <- nTTlumAHER2[which(rownames(nTTlumAHER2) %in% names(nfdradjusted_pvalslumAHER2sig)),]

ntreatedTT1lumAHER2 <- topTreat(nfitlumAHER2, coef=1, number=dim(lumAHER2)[1], lfc=1)
ntreatedTT1lumAHER2sig <- ntreatedTT1lumAHER2[which(ntreatedTT1lumAHER2$adj.P.Val < 0.05),]
ntreatedTT1lumAHER2sig
ntreatedTT2lumAHER2 <- topTreat(nfitlumAHER2, coef=2, number=dim(lumAHER2)[1], lfc=1)
ntreatedTT2lumAHER2sig <- ntreatedTT2lumAHER2[which(ntreatedTT2lumAHER2$adj.P.Val < 0.05),]
ntreatedTT2lumAHER2sig
ntreatedTT3lumAHER2 <- topTreat(nfitlumAHER2, coef=3, number=dim(lumAHER2)[1], lfc=1)
ntreatedTT3lumAHER2sig <- ntreatedTT3lumAHER2[which(ntreatedTT3lumAHER2$adj.P.Val < 0.05),]
ntreatedTT3lumAHER2sig
ntreatedTT4lumAHER2 <- topTreat(nfitlumAHER2, coef=4, number=dim(lumAHER2)[1], lfc=1)
ntreatedTT4lumAHER2sig <- ntreatedTT4lumAHER2[which(ntreatedTT4lumAHER2$adj.P.Val < 0.05),]
ntreatedTT4lumAHER2sig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumAHER2 <- as.data.frame(model_lumAHER2)
nCPEmodel_lumAHER2$CPE <- (nCPEmodel_lumAHER2$CPE)
nthelm2lumAHER2 <- lm(lumAHER2[17876,] ~ subtype * CPE, data=nCPEmodel_lumAHER2)

plot_model(nthelm2lumAHER2, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumAHER2) == "ENSG00000214688") # this is a gene responsible for ER expression
```

```{r}
normallumAHER2 <- treatedTT2lumAHER2sig[which(rownames(treatedTT2lumAHER2sig) %in% rownames(treatedTT4lumAHER2sig)),]
normallumAHER2 <- normallumAHER2[which(!rownames(normallumAHER2) %in% rownames(ntreatedTT2lumAHER2sig)),]

cancerlumAHER2 <- ntreatedTT2lumAHER2sig[which(rownames(ntreatedTT2lumAHER2sig) %in% rownames(ntreatedTT4lumAHER2sig)),]
cancerlumAHER2 <- cancerlumAHER2[which(!rownames(cancerlumAHER2) %in% rownames(treatedTT2lumAHER2sig)),]
cancerlumAHER2
normallumAHER2 


save(normallumAHER2, cancerlumAHER2, file="lumAHER2_C_N_CPE.Rdata")
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumAHER2sig$status <- ifelse(rownames(treatedTT4lumAHER2sig) %in% rownames(cancerlumAHER2), "Within Cancer Cells", ifelse(rownames(treatedTT4lumAHER2sig) %in% rownames(normallumAHER2), "Within Normal Cells", "Both"))
treatedTT4lumAHER2sig$colour <- ifelse(rownames(treatedTT4lumAHER2sig) %in% rownames(cancerlumAHER2), cb3[2], ifelse(rownames(treatedTT4lumAHER2sig) %in% rownames(normallumAHER2), cb3[3], cb3[1]))
geneslumAHER2 <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumAHER2sig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumAHER2 <- geneslumAHER2[!duplicated(geneslumAHER2$GENEID),]
geneslumAHER2$logFC <- treatedTT4lumAHER2sig[which(rownames(treatedTT4lumAHER2sig) %in% geneslumAHER2$GENEID),][geneslumAHER2$GENEID,]$logFC
geneslumAHER2$status <- treatedTT4lumAHER2sig[which(rownames(treatedTT4lumAHER2sig) %in% geneslumAHER2$GENEID),][geneslumAHER2$GENEID,]$status
lumAHER2sigs <- ggplot(geneslumAHER2, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs HER2-enriched Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumAHER2sigsCPE.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumAHER2 <-geneslumAHER2[which(geneslumAHER2$status != "Both"),]
cancernormalonlylumAHER2graph <- ggplot(cancernormalonlylumAHER2, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs HER2-enriched Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumAHER2_CPE.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XlumAHER2s <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumAHER2))
fitlumAHER2s <- lmFit(as.data.frame(lumAHER2), XlumAHER2s)
fitlumAHER2s <- eBayes(fitlumAHER2s)
TTlumAHER2s <- topTable(fitlumAHER2s, number = dim(lumAHER2)[1])
pvalslumAHER2s <- fitlumAHER2s$p.value
fdradjusted_pvalslumAHER2INTERs <- p.adjust(pvalslumAHER2s[,1], method = "fdr")
fdradjusted_pvalslumAHER2sigINTERs <- fdradjusted_pvalslumAHER2INTERs[which(fdradjusted_pvalslumAHER2INTERs < 0.05)]
length(names(fdradjusted_pvalslumAHER2sigINTERs))
fdradjusted_pvalslumAHER2SUBs <- p.adjust(pvalslumAHER2s[,2], method = "fdr")
fdradjusted_pvalslumAHER2sigSUBs <- fdradjusted_pvalslumAHER2SUBs[which(fdradjusted_pvalslumAHER2SUBs < 0.05)]
length(names(fdradjusted_pvalslumAHER2sigSUBs))
fdradjusted_pvalslumAHER2CPEs <- p.adjust(pvalslumAHER2s[,3], method = "fdr")
fdradjusted_pvalslumAHER2sigCPEs <- fdradjusted_pvalslumAHER2CPEs[which(fdradjusted_pvalslumAHER2CPEs < 0.05)]
length(names(fdradjusted_pvalslumAHER2sigCPEs))
plotSA(fitlumAHER2s, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumAHER2SUBs)

treatedTT1lumAHER2s <- topTreat(fitlumAHER2s, coef=1, number=dim(lumAHER2)[1])
treatedTT1lumAHER2sigs <- treatedTT1lumAHER2s[which(treatedTT1lumAHER2s$adj.P.Val < 0.05),]
treatedTT1lumAHER2sigs
treatedTT2lumAHER2s <- topTreat(fitlumAHER2s, coef=2, number=dim(lumAHER2)[1])
treatedTT2lumAHER2sigs <- treatedTT2lumAHER2s[which(treatedTT2lumAHER2s$adj.P.Val < 0.05),]
treatedTT2lumAHER2sigs
treatedTT3lumAHER2s <- topTreat(fitlumAHER2s, coef=3, number=dim(lumAHER2)[1])
treatedTT3lumAHER2sigs <- treatedTT3lumAHER2s[which(treatedTT3lumAHER2s$adj.P.Val < 0.05),]
treatedTT3lumAHER2sigs


treatedTT2lumAHER2sigs[which(rownames(treatedTT2lumAHER2sigs) %in% rownames(treatedTT4lumAHER2sig)),]
treatedTT2lumAHER2sigs[which(rownames(treatedTT2lumAHER2sigs) %in% rownames(cancerlumAHER2)),]
treatedTT2lumAHER2sigs[which(rownames(treatedTT2lumAHER2sigs) %in% rownames(normallumAHER2)),]
```

```{r}
plot(treatedTT4lumAHER2$logFC, -log10(treatedTT4lumAHER2$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumAHER2sig$logFC, -log10(treatedTT4lumAHER2sig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumAHER2s <- model_lumAHER2
nmodel_lumAHER2s$CPE <- (1- nmodel_lumAHER2s$CPE)
nXlumAHER2s <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumAHER2s))
nfitlumAHER2s <- lmFit(as.data.frame(lumAHER2), nXlumAHER2s)
nfitlumAHER2s <- eBayes(nfitlumAHER2s)
nTTlumAHER2s<- topTable(nfitlumAHER2s, number = dim(lumAHER2)[1])
npvalslumAHER2s <- nfitlumAHER2s$p.value
ntreatedTT1lumAHER2s <- topTreat(nfitlumAHER2s, coef=1, number=dim(lumAHER2)[1])
ntreatedTT1lumAHER2sigs <- ntreatedTT1lumAHER2s[which(ntreatedTT1lumAHER2s$adj.P.Val < 0.05),]
ntreatedTT1lumAHER2sigs
ntreatedTT2lumAHER2s <- topTreat(nfitlumAHER2s, coef=2, number=dim(lumAHER2)[1])
ntreatedTT2lumAHER2sigs <- ntreatedTT2lumAHER2s[which(ntreatedTT2lumAHER2s$adj.P.Val < 0.05),]
ntreatedTT2lumAHER2sigs
#ntreatedTT2lumAHER2sigs[which(rownames(ntreatedTT2lumAHER2sigs) %in% rownames(normallumAHER2)),]
ntreatedTT3lumAHER2s <- topTreat(nfitlumAHER2s, coef=3, number=dim(lumAHER2)[1])
ntreatedTT3lumAHER2sigs <- ntreatedTT3lumAHER2s[which(ntreatedTT3lumAHER2s$adj.P.Val < 0.05),]
ntreatedTT3lumAHER2sigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumAHER2s <- as.data.frame(nmodel_lumAHER2s)
nCPEmodel_lumAHER2s$CPE <- (1- nCPEmodel_lumAHER2s$CPE)
nthelm2lumAHER2s <- lm(lumAHER2[5015,] ~ subtype + CPE, data=nCPEmodel_lumAHER2s)

plot_model(nthelm2lumAHER2s, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumAHER2) == "ENSG00000122566")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2lumAHER2s$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2lumAHER2s$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal A", "HER2-enriched")
CPEmodel_lumAHER2 <- as.data.frame(model_lumAHER2)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumAHER2sig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumAHER2 <- lumAHER2[which(rownames(lumAHER2) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumAHER2[top10Genes$GENEID,]
top10GeneslumAHER2 <- top10GeneslumAHER2[order(rownames(top10GeneslumAHER2)),]
dim(top10GeneslumAHER2)
all.equal(rownames(top10GeneslumAHER2), top10Genes$GENEID)
rownames(top10GeneslumAHER2) <- top10Genes$SYMBOL
top10lmlumAHER2 <-  numeric(length = nrow(top10GeneslumAHER2))
plotlumAHER2CPE1 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE1, file=paste0("~/Documents/project/plots/lumAHER2CPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAHER2CPE2 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE2, file=paste0("~/Documents/project/plots/lumAHER2CPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAHER2CPE3 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE3, file=paste0("~/Documents/project/plots/lumAHER2CPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAHER2CPE4 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE4, file=paste0("~/Documents/project/plots/lumAHER2CPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAHER2CPE5 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE5, file=paste0("~/Documents/project/plots/lumAHER2CPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAHER2CPE6 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE6, file=paste0("~/Documents/project/plots/lumAHER2CPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAHER2CPE7 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE7, file=paste0("~/Documents/project/plots/lumAHER2CPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAHER2CPE8 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE8, file=paste0("~/Documents/project/plots/lumAHER2CPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAHER2CPE9 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAHER2)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE9, file=paste0("~/Documents/project/plots/lumAHER2CPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAHER2CPE10 <- ggplot(CPEmodel_lumAHER2 ,aes(y=top10GeneslumAHER2[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumAHER2)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAHER2CPE10, file=paste0("~/Documents/project/plots/lumAHER2CPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumAHER2 <-cowplot::plot_grid(plotlumAHER2CPE1, plotlumAHER2CPE2 + theme(legend.position="none"), plotlumAHER2CPE3 + theme(legend.position="none"),plotlumAHER2CPE4 + theme(legend.position="none"), plotlumAHER2CPE5 + theme(legend.position="none"), plotlumAHER2CPE6 + theme(legend.position="none"),plotlumAHER2CPE7 + theme(legend.position="none"), plotlumAHER2CPE8 + theme(legend.position="none"), plotlumAHER2CPE9 + theme(legend.position="none"),plotlumAHER2CPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumAHER2, file=paste0("~/Documents/project/plots/mergedCPElumAHER2.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XlumAHER2ss <- model.matrix(~ subtype, data=as.data.frame(model_lumAHER2))
fitlumAHER2ss <- lmFit(as.data.frame(lumAHER2), XlumAHER2ss)

fitlumAHER2ss <- eBayes(fitlumAHER2ss)
TTlumAHER2ss <- topTable(fitlumAHER2ss, number = dim(lumAHER2)[1])
pvalslumAHER2ss <- fitlumAHER2ss$p.value
fdradjusted_pvalslumAHER2INTERss <- p.adjust(pvalslumAHER2ss[,1], method = "fdr")
fdradjusted_pvalslumAHER2sigINTERss <- fdradjusted_pvalslumAHER2INTERss[which(fdradjusted_pvalslumAHER2INTERss < 0.05)]
length(names(fdradjusted_pvalslumAHER2sigINTERss))
plotSA(fitlumAHER2ss, main="Final model: Mean-variance trend")

treatedTT1lumAHER2ss <- topTreat(fitlumAHER2ss, coef=1, number=dim(lumAHER2)[1], lfc=1)
treatedTT1lumAHER2sigss <- treatedTT1lumAHER2ss[which(treatedTT1lumAHER2ss$adj.P.Val < 0.05),]
treatedTT1lumAHER2sigss
```

```{r}
plot(treatedTT1lumAHER2ss$logFC, -log10(treatedTT1lumAHER2ss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1lumAHER2sigss$logFC, -log10(treatedTT1lumAHER2sigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4lumAHER2$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4lumAHER2$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4lumAHER2$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumAHER2$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumAHER2$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumAHER2$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumAHER2$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumAHER2$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumAHER2$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumAHER2$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumAHER2s$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumAHER2s$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumAHER2s$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumAHER2s$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumAHER2s$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumAHER2s$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumAHER2sigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1lumAHER2s$adj.P.Val) / length(ntreatedTT1lumAHER2s$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2lumAHER2s$adj.P.Val) / length(ntreatedTT2lumAHER2s$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3lumAHER2s$adj.P.Val) / length(ntreatedTT3lumAHER2s$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1lumAHER2ss$adj.P.Val) / length(treatedTT1lumAHER2ss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumAHER2$adj.P.Val) / length(ntreatedTT4lumAHER2$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2lumAHER2$adj.P.Val) / length(ntreatedTT2lumAHER2$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3lumAHER2$adj.P.Val) / length(ntreatedTT3lumAHER2$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1lumAHER2$adj.P.Val) / length(ntreatedTT1lumAHER2$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumAHER2$adj.P.Val) / length(ntreatedTT1lumAHER2$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumAHER2$adj.P.Val) / length(ntreatedTT4lumAHER2$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumAHER2$adj.P.Val) / length(ntreatedTT2lumAHER2$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumAHER2$adj.P.Val) / length(ntreatedTT3lumAHER2$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumAHER2$adj.P.Val) / length(treatedTT1lumAHER2$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumAHER2$adj.P.Val) / length(treatedTT3lumAHER2$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumAHER2$adj.P.Val) / length(treatedTT4lumAHER2$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumAHER2$adj.P.Val) / length(treatedTT2lumAHER2$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT2lumAHER2$adj.P.Val) / length(treatedTT2lumAHER2$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumAHER2$adj.P.Val) / length(ntreatedTT2lumAHER2$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumAHER2$adj.P.Val) / length(treatedTT3lumAHER2$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumAHER2$adj.P.Val) / length(ntreatedTT3lumAHER2$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumAHER2$adj.P.Val) / length(treatedTT4lumAHER2$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumAHER2$adj.P.Val) / length(ntreatedTT4lumAHER2$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```




