
```{r, include = FALSE} 
library(ggplot2)
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }
load(file="~/Documents/project/normaliseddataCPE.Rdata")
lumAlumB <- datalumAlumB
load(file="~/Documents/project/modelmatricesCPE.Rdata")
```



read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XlumAlumB <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumAlumB))
fitlumAlumB <- lmFit(as.data.frame(lumAlumB), XlumAlumB)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XlumAlumB))
#contr.matrix
#contrastsfitlumAlumB <- contrasts.fit(fitlumAlumB, contrasts=contr.matrix)
#contrastsfitlumAlumB <- eBayes(contrastsfitlumAlumB)
fitlumAlumB <- eBayes(fitlumAlumB)
TTlumAlumB <- topTable(fitlumAlumB, number = dim(lumAlumB)[1])
pvalslumAlumB <- fitlumAlumB$p.value
fdradjusted_pvalslumAlumB <- p.adjust(pvalslumAlumB[,4], method = "fdr")
fdradjusted_pvalslumAlumBsig <- fdradjusted_pvalslumAlumB[which(fdradjusted_pvalslumAlumB < 0.05)]
names(fdradjusted_pvalslumAlumBsig)
top95lumAlumB <- TTlumAlumB[which(rownames(TTlumAlumB) %in% names(fdradjusted_pvalslumAlumBsig)),]
#plotMA(treatedTT4lumAlumB , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTlumAlumB$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitlumAlumB, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumAlumB)
treatedTT1lumAlumB <- topTreat(fitlumAlumB, coef=1, number=dim(lumAlumB)[1], lfc=1)
treatedTT1lumAlumBsig <- treatedTT1lumAlumB[which(treatedTT1lumAlumB$adj.P.Val < 0.05),]
treatedTT1lumAlumBsig
treatedTT2lumAlumB <- topTreat(fitlumAlumB, coef=2, number=dim(lumAlumB)[1], lfc=1)
treatedTT2lumAlumBsig <- treatedTT2lumAlumB[which(treatedTT2lumAlumB$adj.P.Val < 0.05),]
treatedTT2lumAlumBsig
treatedTT3lumAlumB <- topTreat(fitlumAlumB, coef=3, number=dim(lumAlumB)[1], lfc=1)
treatedTT3lumAlumBsig <- treatedTT3lumAlumB[which(treatedTT3lumAlumB$adj.P.Val < 0.05),]
treatedTT3lumAlumBsig
treatedTT4lumAlumB <- topTreat(fitlumAlumB, coef=4, number=dim(lumAlumB)[1], lfc=1)
treatedTT4lumAlumBsig <- treatedTT4lumAlumB[which(treatedTT4lumAlumB$adj.P.Val < 0.05),]
treatedTT4lumAlumBsig
#plot(density(treatedTT2lumAlumBsig$logFC))
#ENSG00000122566
#ENSG00000122566


```






```{r}
plot(treatedTT4lumAlumB$logFC, -log10(treatedTT4lumAlumB$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumAlumBsig$logFC, -log10(treatedTT4lumAlumBsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumAlumB <-model_lumAlumB
nmodel_lumAlumB$CPE <- (1- nmodel_lumAlumB$CPE)
nXlumAlumB <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumAlumB))
nfitlumAlumB <- lmFit(as.data.frame(lumAlumB), nXlumAlumB)

nfitlumAlumB <- eBayes(nfitlumAlumB)
nTTlumAlumB<- topTable(nfitlumAlumB, number = dim(lumAlumB)[1])
npvalslumAlumB <- nfitlumAlumB$p.value
nfdradjusted_pvalslumAlumB <- p.adjust(npvalslumAlumB[,4], method = "fdr")
nfdradjusted_pvalslumAlumBsig <- nfdradjusted_pvalslumAlumB[which(nfdradjusted_pvalslumAlumB < 0.05)]
names(nfdradjusted_pvalslumAlumBsig)
ntop95lumAlumB <- nTTlumAlumB[which(rownames(nTTlumAlumB) %in% names(nfdradjusted_pvalslumAlumBsig)),]

ntreatedTT1lumAlumB <- topTreat(nfitlumAlumB, coef=1, number=dim(lumAlumB)[1], lfc=1)
ntreatedTT1lumAlumBsig <- ntreatedTT1lumAlumB[which(ntreatedTT1lumAlumB$adj.P.Val < 0.05),]
ntreatedTT1lumAlumBsig
ntreatedTT2lumAlumB <- topTreat(nfitlumAlumB, coef=2, number=dim(lumAlumB)[1], lfc=1)
ntreatedTT2lumAlumBsig <- ntreatedTT2lumAlumB[which(ntreatedTT2lumAlumB$adj.P.Val < 0.05),]
ntreatedTT2lumAlumBsig
ntreatedTT3lumAlumB <- topTreat(nfitlumAlumB, coef=3, number=dim(lumAlumB)[1], lfc=1)
ntreatedTT3lumAlumBsig <- ntreatedTT3lumAlumB[which(ntreatedTT3lumAlumB$adj.P.Val < 0.05),]
ntreatedTT3lumAlumBsig
ntreatedTT4lumAlumB <- topTreat(nfitlumAlumB, coef=4, number=dim(lumAlumB)[1], lfc=1)
ntreatedTT4lumAlumBsig <- ntreatedTT4lumAlumB[which(ntreatedTT4lumAlumB$adj.P.Val < 0.05),]
ntreatedTT4lumAlumBsig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumAlumB <- as.data.frame(model_lumAlumB)
nCPEmodel_lumAlumB$CPE <- (nCPEmodel_lumAlumB$CPE)
nthelm2lumAlumB <- lm(lumAlumB[5098,] ~ subtype * CPE, data=nCPEmodel_lumAlumB)

plot_model(nthelm2lumAlumB, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumAlumB) == "ENSG00000122566") # this is a gene responsible for ER expression
```

```{r}
normallumAlumB <- treatedTT2lumAlumBsig[which(rownames(treatedTT2lumAlumBsig) %in% rownames(treatedTT4lumAlumBsig)),]
normallumAlumB <- normallumAlumB[which(!rownames(normallumAlumB) %in% rownames(ntreatedTT2lumAlumBsig)),]

cancerlumAlumB <- ntreatedTT2lumAlumBsig[which(rownames(ntreatedTT2lumAlumBsig) %in% rownames(ntreatedTT4lumAlumBsig)),]
cancerlumAlumB <- cancerlumAlumB[which(!rownames(cancerlumAlumB) %in% rownames(treatedTT2lumAlumBsig)),]
cancerlumAlumB
normallumAlumB 


save(normallumAlumB, cancerlumAlumB, file="lumAlumB_C_N_CPE.Rdata")
ntreatedTT4lumAlumBsig
jv <- ntreatedTT4lumAlumBsig[which(!rownames(ntreatedTT4lumAlumBsig) %in% rownames(normallumAlumB)),]
jv[which(!rownames(jv) %in% rownames(cancerlumAlumB)),]
534-467-15
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumAlumBsig$status <- ifelse(rownames(treatedTT4lumAlumBsig) %in% rownames(cancerlumAlumB), "Within Cancer Cells", ifelse(rownames(treatedTT4lumAlumBsig) %in% rownames(normallumAlumB), "Within Normal Cells", "Both"))
treatedTT4lumAlumBsig$colour <- ifelse(rownames(treatedTT4lumAlumBsig) %in% rownames(cancerlumAlumB), cb3[2], ifelse(rownames(treatedTT4lumAlumBsig) %in% rownames(normallumAlumB), cb3[3], cb3[1]))
geneslumAlumB <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumAlumBsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumAlumB <- geneslumAlumB[!duplicated(geneslumAlumB$GENEID),]
geneslumAlumB$logFC <- treatedTT4lumAlumBsig[which(rownames(treatedTT4lumAlumBsig) %in% geneslumAlumB$GENEID),][geneslumAlumB$GENEID,]$logFC
geneslumAlumB$status <- treatedTT4lumAlumBsig[which(rownames(treatedTT4lumAlumBsig) %in% geneslumAlumB$GENEID),][geneslumAlumB$GENEID,]$status
lumAlumBsigs <- ggplot(geneslumAlumB, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Luminal B Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumAlumBsigsCPE.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumAlumB <-geneslumAlumB[which(geneslumAlumB$status != "Both"),]
cancernormalonlylumAlumBgraph <- ggplot(cancernormalonlylumAlumB, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal A vs Luminal B Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumAlumBCPE.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XlumAlumBs <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumAlumB))
fitlumAlumBs <- lmFit(as.data.frame(lumAlumB), XlumAlumBs)
fitlumAlumBs <- eBayes(fitlumAlumBs)
TTlumAlumBs <- topTable(fitlumAlumBs, number = dim(lumAlumB)[1])
pvalslumAlumBs <- fitlumAlumBs$p.value
fdradjusted_pvalslumAlumBINTERs <- p.adjust(pvalslumAlumBs[,1], method = "fdr")
fdradjusted_pvalslumAlumBsigINTERs <- fdradjusted_pvalslumAlumBINTERs[which(fdradjusted_pvalslumAlumBINTERs < 0.05)]
length(names(fdradjusted_pvalslumAlumBsigINTERs))
fdradjusted_pvalslumAlumBSUBs <- p.adjust(pvalslumAlumBs[,2], method = "fdr")
fdradjusted_pvalslumAlumBsigSUBs <- fdradjusted_pvalslumAlumBSUBs[which(fdradjusted_pvalslumAlumBSUBs < 0.05)]
length(names(fdradjusted_pvalslumAlumBsigSUBs))
fdradjusted_pvalslumAlumBCPEs <- p.adjust(pvalslumAlumBs[,3], method = "fdr")
fdradjusted_pvalslumAlumBsigCPEs <- fdradjusted_pvalslumAlumBCPEs[which(fdradjusted_pvalslumAlumBCPEs < 0.05)]
length(names(fdradjusted_pvalslumAlumBsigCPEs))
plotSA(fitlumAlumBs, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumAlumBSUBs)

treatedTT1lumAlumBs <- topTreat(fitlumAlumBs, coef=1, number=dim(lumAlumB)[1])
treatedTT1lumAlumBsigs <- treatedTT1lumAlumBs[which(treatedTT1lumAlumBs$adj.P.Val < 0.05),]
treatedTT1lumAlumBsigs
treatedTT2lumAlumBs <- topTreat(fitlumAlumBs, coef=2, number=dim(lumAlumB)[1])
treatedTT2lumAlumBsigs <- treatedTT2lumAlumBs[which(treatedTT2lumAlumBs$adj.P.Val < 0.05),]
treatedTT2lumAlumBsigs
treatedTT3lumAlumBs <- topTreat(fitlumAlumBs, coef=3, number=dim(lumAlumB)[1])
treatedTT3lumAlumBsigs <- treatedTT3lumAlumBs[which(treatedTT3lumAlumBs$adj.P.Val < 0.05),]
treatedTT3lumAlumBsigs


treatedTT2lumAlumBsigs[which(rownames(treatedTT2lumAlumBsigs) %in% rownames(treatedTT4lumAlumBsig)),]
treatedTT2lumAlumBsigs[which(rownames(treatedTT2lumAlumBsigs) %in% rownames(cancerlumAlumB)),]
treatedTT2lumAlumBsigs[which(rownames(treatedTT2lumAlumBsigs) %in% rownames(normallumAlumB)),]
```

```{r}
plot(treatedTT4lumAlumB$logFC, -log10(treatedTT4lumAlumB$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumAlumBsig$logFC, -log10(treatedTT4lumAlumBsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumAlumBs <- model_lumAlumB
nmodel_lumAlumBs$CPE <- (1- nmodel_lumAlumBs$CPE)
nXlumAlumBs <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumAlumBs))
nfitlumAlumBs <- lmFit(as.data.frame(lumAlumB), nXlumAlumBs)
nfitlumAlumBs <- eBayes(nfitlumAlumBs)
nTTlumAlumBs<- topTable(nfitlumAlumBs, number = dim(lumAlumB)[1])
npvalslumAlumBs <- nfitlumAlumBs$p.value
ntreatedTT1lumAlumBs <- topTreat(nfitlumAlumBs, coef=1, number=dim(lumAlumB)[1])
ntreatedTT1lumAlumBsigs <- ntreatedTT1lumAlumBs[which(ntreatedTT1lumAlumBs$adj.P.Val < 0.05),]
ntreatedTT1lumAlumBsigs
ntreatedTT2lumAlumBs <- topTreat(nfitlumAlumBs, coef=2, number=dim(lumAlumB)[1])
ntreatedTT2lumAlumBsigs <- ntreatedTT2lumAlumBs[which(ntreatedTT2lumAlumBs$adj.P.Val < 0.05),]
ntreatedTT2lumAlumBsigs
#ntreatedTT2lumAlumBsigs[which(rownames(ntreatedTT2lumAlumBsigs) %in% rownames(normallumAlumB)),]
ntreatedTT3lumAlumBs <- topTreat(nfitlumAlumBs, coef=3, number=dim(lumAlumB)[1])
ntreatedTT3lumAlumBsigs <- ntreatedTT3lumAlumBs[which(ntreatedTT3lumAlumBs$adj.P.Val < 0.05),]
ntreatedTT3lumAlumBsigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumAlumBs <- as.data.frame(nmodel_lumAlumBs)
nCPEmodel_lumAlumBs$CPE <- (1-nCPEmodel_lumAlumBs$CPE)
nthelm2lumAlumBs <- lm(lumAlumB[ 3767,] ~ subtype * CPE, data=nCPEmodel_lumAlumBs)

plot_model(nthelm2lumAlumBs, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumAlumB) == "ENSG00000111404")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2lumAlumBs$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2lumAlumBs$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal A", "Luminal B")
CPEmodel_lumAlumB <- as.data.frame(model_lumAlumB)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumAlumBsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumAlumB <- lumAlumB[which(rownames(lumAlumB) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumAlumB[top10Genes$GENEID,]
top10GeneslumAlumB <- top10GeneslumAlumB[order(rownames(top10GeneslumAlumB)),]
dim(top10GeneslumAlumB)
all.equal(rownames(top10GeneslumAlumB), top10Genes$GENEID)
rownames(top10GeneslumAlumB) <- top10Genes$SYMBOL
top10lmlumAlumB <-  numeric(length = nrow(top10GeneslumAlumB))
plotlumAlumBCPE1 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE1, file=paste0("~/Documents/project/plots/lumAlumBCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAlumBCPE2 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE2, file=paste0("~/Documents/project/plots/lumAlumBCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAlumBCPE3 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE3, file=paste0("~/Documents/project/plots/lumAlumBCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAlumBCPE4 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE4, file=paste0("~/Documents/project/plots/lumAlumBCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAlumBCPE5 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE5, file=paste0("~/Documents/project/plots/lumAlumBCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAlumBCPE6 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE6, file=paste0("~/Documents/project/plots/lumAlumBCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAlumBCPE7 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE7, file=paste0("~/Documents/project/plots/lumAlumBCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAlumBCPE8 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE8, file=paste0("~/Documents/project/plots/lumAlumBCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumAlumBCPE9 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumAlumB)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE9, file=paste0("~/Documents/project/plots/lumAlumBCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumAlumBCPE10 <- ggplot(CPEmodel_lumAlumB ,aes(y=top10GeneslumAlumB[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumAlumB)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumAlumBCPE10, file=paste0("~/Documents/project/plots/lumAlumBCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumAlumB <-cowplot::plot_grid(plotlumAlumBCPE1, plotlumAlumBCPE2 + theme(legend.position="none"), plotlumAlumBCPE3 + theme(legend.position="none"),plotlumAlumBCPE4 + theme(legend.position="none"), plotlumAlumBCPE5 + theme(legend.position="none"), plotlumAlumBCPE6 + theme(legend.position="none"),plotlumAlumBCPE7 + theme(legend.position="none"), plotlumAlumBCPE8 + theme(legend.position="none"), plotlumAlumBCPE9 + theme(legend.position="none"),plotlumAlumBCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumAlumB, file=paste0("~/Documents/project/plots/mergedCPElumAlumB.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumAlumB <-cowplot::plot_grid(plotlumAlumBCPE1, plotlumAlumBCPE2 + theme(legend.position="none"), plotlumAlumBCPE3 + theme(legend.position="none"),plotlumAlumBCPE4 + theme(legend.position="none"), plotlumAlumBCPE5 + theme(legend.position="none"), plotlumAlumBCPE6 + theme(legend.position="none"),plotlumAlumBCPE7 + theme(legend.position="none"), plotlumAlumBCPE8 + theme(legend.position="none"), plotlumAlumBCPE9 + theme(legend.position="none"),plotlumAlumBCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumAlumB, file=paste0("~/Documents/project/plots/mergedCPElumAlumB.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumAlumBCPEbarplots <-cowplot::plot_grid(lumAHER2sigs, lumANormalsigs + theme(legend.position="none"), lumABasalsigs + theme(legend.position="none"),lumBBasalsigs + theme(legend.position="none"), align = "hv", ncol=2)

ggsave(mergedCPElumAlumBCPEbarplots, file=paste0("~/Documents/project/plots/mergedCPElumAlumBCPEbarplots.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XlumAlumBss <- model.matrix(~ subtype, data=as.data.frame(model_lumAlumB))
fitlumAlumBss <- lmFit(as.data.frame(lumAlumB), XlumAlumBss)

fitlumAlumBss <- eBayes(fitlumAlumBss)
TTlumAlumBss <- topTable(fitlumAlumBss, number = dim(lumAlumB)[1])
pvalslumAlumBss <- fitlumAlumBss$p.value
fdradjusted_pvalslumAlumBINTERss <- p.adjust(pvalslumAlumBss[,1], method = "fdr")
fdradjusted_pvalslumAlumBsigINTERss <- fdradjusted_pvalslumAlumBINTERss[which(fdradjusted_pvalslumAlumBINTERss < 0.05)]
length(names(fdradjusted_pvalslumAlumBsigINTERss))
plotSA(fitlumAlumBss, main="Final model: Mean-variance trend")

treatedTT1lumAlumBss <- topTreat(fitlumAlumBss, coef=1, number=dim(lumAlumB)[1], lfc=1)
treatedTT1lumAlumBsigss <- treatedTT1lumAlumBss[which(treatedTT1lumAlumBss$adj.P.Val < 0.05),]
treatedTT1lumAlumBsigss
```

```{r}
plot(treatedTT1lumAlumBss$logFC, -log10(treatedTT1lumAlumBss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1lumAlumBsigss$logFC, -log10(treatedTT1lumAlumBsigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4lumAlumB$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4lumAlumB$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4lumAlumB$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumAlumB$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumAlumB$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumAlumB$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumAlumB$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumAlumB$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumAlumB$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumAlumB$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumAlumBs$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumAlumBs$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumAlumBs$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumAlumBs$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumAlumBs$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumAlumBs$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumAlumBsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1lumAlumBs$adj.P.Val) / length(ntreatedTT1lumAlumBs$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2lumAlumBs$adj.P.Val) / length(ntreatedTT2lumAlumBs$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3lumAlumBs$adj.P.Val) / length(ntreatedTT3lumAlumBs$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1lumAlumBss$adj.P.Val) / length(treatedTT1lumAlumBss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumAlumB$adj.P.Val) / length(ntreatedTT4lumAlumB$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2lumAlumB$adj.P.Val) / length(ntreatedTT2lumAlumB$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3lumAlumB$adj.P.Val) / length(ntreatedTT3lumAlumB$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1lumAlumB$adj.P.Val) / length(ntreatedTT1lumAlumB$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumAlumB$adj.P.Val) / length(ntreatedTT1lumAlumB$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumAlumB$adj.P.Val) / length(ntreatedTT4lumAlumB$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumAlumB$adj.P.Val) / length(ntreatedTT2lumAlumB$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumAlumB$adj.P.Val) / length(ntreatedTT3lumAlumB$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumAlumB$adj.P.Val) / length(treatedTT1lumAlumB$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumAlumB$adj.P.Val) / length(treatedTT3lumAlumB$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumAlumB$adj.P.Val) / length(treatedTT4lumAlumB$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumAlumB$adj.P.Val) / length(treatedTT2lumAlumB$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2lumAlumB$adj.P.Val) / length(treatedTT2lumAlumB$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumAlumB$adj.P.Val) / length(ntreatedTT2lumAlumB$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumAlumB$adj.P.Val) / length(treatedTT3lumAlumB$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumAlumB$adj.P.Val) / length(ntreatedTT3lumAlumB$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumAlumB$adj.P.Val) / length(treatedTT4lumAlumB$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumAlumB$adj.P.Val) / length(ntreatedTT4lumAlumB$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```


