
```{r, include = FALSE} 
library(ggplot2)#7 rows
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }

load("~/Documents/project/normaliseddataCPE.Rdata")
load("~/Documents/project/modelmatricesCPE.Rdata")

#load(file="~/Documents/project/normaliseddata.Rdata")
lumBBasal <- datalumBBasal
#load(file="~/Documents/project/modelmatrices.Rdata")
```




read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XlumBBasal <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumBBasal))
fitlumBBasal <- lmFit(as.data.frame(lumBBasal), XlumBBasal)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XlumBBasal))
#contr.matrix
#contrastsfitlumBBasal <- contrasts.fit(fitlumBBasal, contrasts=contr.matrix)
#contrastsfitlumBBasal <- eBayes(contrastsfitlumBBasal)
fitlumBBasal <- eBayes(fitlumBBasal)
TTlumBBasal <- topTable(fitlumBBasal, number = dim(lumBBasal)[1])
pvalslumBBasal <- fitlumBBasal$p.value
fdradjusted_pvalslumBBasal <- p.adjust(pvalslumBBasal[,4], method = "fdr")
fdradjusted_pvalslumBBasalsig <- fdradjusted_pvalslumBBasal[which(fdradjusted_pvalslumBBasal < 0.05)]
names(fdradjusted_pvalslumBBasalsig)
top95lumBBasal <- TTlumBBasal[which(rownames(TTlumBBasal) %in% names(fdradjusted_pvalslumBBasalsig)),]
#plotMA(treatedTT4lumBBasal , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTlumBBasal$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitlumBBasal, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumBBasal)
treatedTT1lumBBasal <- topTreat(fitlumBBasal, coef=1, number=dim(lumBBasal)[1], lfc=1)
treatedTT1lumBBasalsig <- treatedTT1lumBBasal[which(treatedTT1lumBBasal$adj.P.Val < 0.05),]
treatedTT1lumBBasalsig
treatedTT2lumBBasal <- topTreat(fitlumBBasal, coef=2, number=dim(lumBBasal)[1], lfc=1)
treatedTT2lumBBasalsig <- treatedTT2lumBBasal[which(treatedTT2lumBBasal$adj.P.Val < 0.05),]
treatedTT2lumBBasalsig
treatedTT3lumBBasal <- topTreat(fitlumBBasal, coef=3, number=dim(lumBBasal)[1], lfc=1)
treatedTT3lumBBasalsig <- treatedTT3lumBBasal[which(treatedTT3lumBBasal$adj.P.Val < 0.05),]
treatedTT3lumBBasalsig
treatedTT4lumBBasal <- topTreat(fitlumBBasal, coef=4, number=dim(lumBBasal)[1], lfc=1)
treatedTT4lumBBasalsig <- treatedTT4lumBBasal[which(treatedTT4lumBBasal$adj.P.Val < 0.05),]
treatedTT4lumBBasalsig
#plot(density(treatedTT2lumBBasalsig$logFC))
#ENSG00000122566
#ENSG00000122566

save(treatedTT4lumBBasalsig, treatedTT2lumBBasalsig, file="lumBBasaloutput.Rdata")
```






```{r}
plot(treatedTT4lumBBasal$logFC, -log10(treatedTT4lumBBasal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumBBasalsig$logFC, -log10(treatedTT4lumBBasalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumBBasal <-model_lumBBasal
nmodel_lumBBasal$CPE <- (1- nmodel_lumBBasal$CPE)
nXlumBBasal <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumBBasal))
nfitlumBBasal <- lmFit(as.data.frame(lumBBasal), nXlumBBasal)

nfitlumBBasal <- eBayes(nfitlumBBasal)
nTTlumBBasal<- topTable(nfitlumBBasal, number = dim(lumBBasal)[1])
npvalslumBBasal <- nfitlumBBasal$p.value
nfdradjusted_pvalslumBBasal <- p.adjust(npvalslumBBasal[,4], method = "fdr")
nfdradjusted_pvalslumBBasalsig <- nfdradjusted_pvalslumBBasal[which(nfdradjusted_pvalslumBBasal < 0.05)]
names(nfdradjusted_pvalslumBBasalsig)
ntop95lumBBasal <- nTTlumBBasal[which(rownames(nTTlumBBasal) %in% names(nfdradjusted_pvalslumBBasalsig)),]

ntreatedTT1lumBBasal <- topTreat(nfitlumBBasal, coef=1, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT1lumBBasalsig <- ntreatedTT1lumBBasal[which(ntreatedTT1lumBBasal$adj.P.Val < 0.05),]
ntreatedTT1lumBBasalsig
ntreatedTT2lumBBasal <- topTreat(nfitlumBBasal, coef=2, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT2lumBBasalsig <- ntreatedTT2lumBBasal[which(ntreatedTT2lumBBasal$adj.P.Val < 0.05),]
ntreatedTT2lumBBasalsig
ntreatedTT3lumBBasal <- topTreat(nfitlumBBasal, coef=3, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT3lumBBasalsig <- ntreatedTT3lumBBasal[which(ntreatedTT3lumBBasal$adj.P.Val < 0.05),]
ntreatedTT3lumBBasalsig
ntreatedTT4lumBBasal <- topTreat(nfitlumBBasal, coef=4, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT4lumBBasalsig <- ntreatedTT4lumBBasal[which(ntreatedTT4lumBBasal$adj.P.Val < 0.05),]
ntreatedTT4lumBBasalsig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumBBasal <- as.data.frame(model_lumBBasal)
nCPEmodel_lumBBasal$CPE <- (nCPEmodel_lumBBasal$CPE)
nthelm2lumBBasal <- lm(lumBBasal[17876,] ~ subtype * CPE, data=nCPEmodel_lumBBasal)

plot_model(nthelm2lumBBasal, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumBBasal) == "ENSG00000214688") # this is a gene responsible for ER expression
```

```{r}
normallumBBasal <- treatedTT2lumBBasalsig[which(rownames(treatedTT2lumBBasalsig) %in% rownames(treatedTT4lumBBasalsig)),]
normallumBBasal <- normallumBBasal[which(!rownames(normallumBBasal) %in% rownames(ntreatedTT2lumBBasalsig)),]

cancerlumBBasal <- ntreatedTT2lumBBasalsig[which(rownames(ntreatedTT2lumBBasalsig) %in% rownames(ntreatedTT4lumBBasalsig)),]
cancerlumBBasal <- cancerlumBBasal[which(!rownames(cancerlumBBasal) %in% rownames(treatedTT2lumBBasalsig)),]
cancerlumBBasal
normallumBBasal 
save(normallumBBasal, cancerlumBBasal, file="lumBBasal_C_N_CPE.Rdata")
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumBBasalsig$status <- ifelse(rownames(treatedTT4lumBBasalsig) %in% rownames(cancerlumBBasal), "Within Cancer Cells", ifelse(rownames(treatedTT4lumBBasalsig) %in% rownames(normallumBBasal), "Within Normal Cells", "Both"))
treatedTT4lumBBasalsig$colour <- ifelse(rownames(treatedTT4lumBBasalsig) %in% rownames(cancerlumBBasal), cb3[2], ifelse(rownames(treatedTT4lumBBasalsig) %in% rownames(normallumBBasal), cb3[3], cb3[1]))
geneslumBBasal <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumBBasalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumBBasal <- geneslumBBasal[!duplicated(geneslumBBasal$GENEID),]
geneslumBBasal <- geneslumBBasal[!duplicated(geneslumBBasal$SYMBOL),]
geneslumBBasal$logFC <- treatedTT4lumBBasalsig[which(rownames(treatedTT4lumBBasalsig) %in% geneslumBBasal$GENEID),][geneslumBBasal$GENEID,]$logFC
geneslumBBasal$status <- treatedTT4lumBBasalsig[which(rownames(treatedTT4lumBBasalsig) %in% geneslumBBasal$GENEID),][geneslumBBasal$GENEID,]$status
which(duplicated(geneslumBBasal$SYMBOL))
lumBBasalsigs <- ggplot(geneslumBBasal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal B vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumBBasalsigs.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumBBasal <-geneslumBBasal[which(geneslumBBasal$status != "Both"),]
cancernormalonlylumBBasalgraph <- ggplot(cancernormalonlylumBBasal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal B vs Basal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumBBasal.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XlumBBasals <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumBBasal))
fitlumBBasals <- lmFit(as.data.frame(lumBBasal), XlumBBasals)
fitlumBBasals <- eBayes(fitlumBBasals)
TTlumBBasals <- topTable(fitlumBBasals, number = dim(lumBBasal)[1])
pvalslumBBasals <- fitlumBBasals$p.value
fdradjusted_pvalslumBBasalINTERs <- p.adjust(pvalslumBBasals[,1], method = "fdr")
fdradjusted_pvalslumBBasalsigINTERs <- fdradjusted_pvalslumBBasalINTERs[which(fdradjusted_pvalslumBBasalINTERs < 0.05)]
length(names(fdradjusted_pvalslumBBasalsigINTERs))
fdradjusted_pvalslumBBasalSUBs <- p.adjust(pvalslumBBasals[,2], method = "fdr")
fdradjusted_pvalslumBBasalsigSUBs <- fdradjusted_pvalslumBBasalSUBs[which(fdradjusted_pvalslumBBasalSUBs < 0.05)]
length(names(fdradjusted_pvalslumBBasalsigSUBs))
fdradjusted_pvalslumBBasalCPEs <- p.adjust(pvalslumBBasals[,3], method = "fdr")
fdradjusted_pvalslumBBasalsigCPEs <- fdradjusted_pvalslumBBasalCPEs[which(fdradjusted_pvalslumBBasalCPEs < 0.05)]
length(names(fdradjusted_pvalslumBBasalsigCPEs))
plotSA(fitlumBBasals, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumBBasalSUBs)

treatedTT1lumBBasals <- topTreat(fitlumBBasals, coef=1, number=dim(lumBBasal)[1], lfc=1)
treatedTT1lumBBasalsigs <- treatedTT1lumBBasals[which(treatedTT1lumBBasals$adj.P.Val < 0.05),]
treatedTT1lumBBasalsigs
treatedTT2lumBBasals <- topTreat(fitlumBBasals, coef=2, number=dim(lumBBasal)[1], lfc=1)
treatedTT2lumBBasalsigs <- treatedTT2lumBBasals[which(treatedTT2lumBBasals$adj.P.Val < 0.05),]
treatedTT2lumBBasalsigs
treatedTT3lumBBasals <- topTreat(fitlumBBasals, coef=3, number=dim(lumBBasal)[1], lfc=1)
treatedTT3lumBBasalsigs <- treatedTT3lumBBasals[which(treatedTT3lumBBasals$adj.P.Val < 0.05),]
treatedTT3lumBBasalsigs


treatedTT2lumBBasalsigs[which(rownames(treatedTT2lumBBasalsigs) %in% rownames(treatedTT4lumBBasalsig)),]
treatedTT2lumBBasalsigs[which(rownames(treatedTT2lumBBasalsigs) %in% rownames(cancerlumBBasal)),]
treatedTT2lumBBasalsigs[which(rownames(treatedTT2lumBBasalsigs) %in% rownames(normallumBBasal)),]
```

```{r}
plot(treatedTT4lumBBasal$logFC, -log10(treatedTT4lumBBasal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumBBasalsig$logFC, -log10(treatedTT4lumBBasalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumBBasals <- model_lumBBasal
nmodel_lumBBasals$CPE <- (1- nmodel_lumBBasals$CPE)
nXlumBBasals <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumBBasals))
nfitlumBBasals <- lmFit(as.data.frame(lumBBasal), nXlumBBasals)
nfitlumBBasals <- eBayes(nfitlumBBasals)
nTTlumBBasals<- topTable(nfitlumBBasals, number = dim(lumBBasal)[1])
npvalslumBBasals <- nfitlumBBasals$p.value
ntreatedTT1lumBBasals <- topTreat(nfitlumBBasals, coef=1, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT1lumBBasalsigs <- ntreatedTT1lumBBasals[which(ntreatedTT1lumBBasals$adj.P.Val < 0.05),]
ntreatedTT1lumBBasalsigs
ntreatedTT2lumBBasals <- topTreat(nfitlumBBasals, coef=2, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT2lumBBasalsigs <- ntreatedTT2lumBBasals[which(ntreatedTT2lumBBasals$adj.P.Val < 0.05),]
ntreatedTT2lumBBasalsigs
#ntreatedTT2lumBBasalsigs[which(rownames(ntreatedTT2lumBBasalsigs) %in% rownames(normallumBBasal)),]
ntreatedTT3lumBBasals <- topTreat(nfitlumBBasals, coef=3, number=dim(lumBBasal)[1], lfc=1)
ntreatedTT3lumBBasalsigs <- ntreatedTT3lumBBasals[which(ntreatedTT3lumBBasals$adj.P.Val < 0.05),]
ntreatedTT3lumBBasalsigs
```




single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumBBasals <- as.data.frame(nmodel_lumBBasals)
nCPEmodel_lumBBasals$CPE <- (1- nCPEmodel_lumBBasals$CPE)
nthelm2lumBBasals <- lm(lumBBasal[5015,] ~ subtype + CPE, data=nCPEmodel_lumBBasals)

plot_model(nthelm2lumBBasals, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumBBasal) == "ENSG00000122566")
```

plot adj p values:
```{r}
qqnorm(ntreatedTT2lumBBasals$adj.P.Val, pch = 1, frame = FALSE)
qqline(ntreatedTT2lumBBasals$adj.P.Val, col = "steelblue", lwd = 2)
```


merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal B", "Basal-like")
CPEmodel_lumBBasal <- as.data.frame(model_lumBBasal)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumBBasalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumBBasal <- lumBBasal[which(rownames(lumBBasal) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumBBasal[top10Genes$GENEID,]
top10GeneslumBBasal <- top10GeneslumBBasal[order(rownames(top10GeneslumBBasal)),]
dim(top10GeneslumBBasal)
all.equal(rownames(top10GeneslumBBasal), top10Genes$GENEID)
rownames(top10GeneslumBBasal) <- top10Genes$SYMBOL
top10lmlumBBasal <-  numeric(length = nrow(top10GeneslumBBasal))
plotlumBBasalCPE1 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE1, file=paste0("~/Documents/project/plots/lumBBasalCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBBasalCPE2 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE2, file=paste0("~/Documents/project/plots/lumBBasalCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBBasalCPE3 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE3, file=paste0("~/Documents/project/plots/lumBBasalCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBBasalCPE4 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE4, file=paste0("~/Documents/project/plots/lumBBasalCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBBasalCPE5 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE5, file=paste0("~/Documents/project/plots/lumBBasalCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBBasalCPE6 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE6, file=paste0("~/Documents/project/plots/lumBBasalCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBBasalCPE7 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE7, file=paste0("~/Documents/project/plots/lumBBasalCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBBasalCPE8 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE8, file=paste0("~/Documents/project/plots/lumBBasalCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBBasalCPE9 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBBasal)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE9, file=paste0("~/Documents/project/plots/lumBBasalCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBBasalCPE10 <- ggplot(CPEmodel_lumBBasal ,aes(y=top10GeneslumBBasal[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumBBasal)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBBasalCPE10, file=paste0("~/Documents/project/plots/lumBBasalCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumBBasal <-cowplot::plot_grid(plotlumBBasalCPE1, plotlumBBasalCPE2 + theme(legend.position="none"), plotlumBBasalCPE3 + theme(legend.position="none"),plotlumBBasalCPE4 + theme(legend.position="none"), plotlumBBasalCPE5 + theme(legend.position="none"), plotlumBBasalCPE6 + theme(legend.position="none"),plotlumBBasalCPE7 + theme(legend.position="none"), plotlumBBasalCPE8 + theme(legend.position="none"), plotlumBBasalCPE9 + theme(legend.position="none"),plotlumBBasalCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumBBasal, file=paste0("~/Documents/project/plots/mergedCPElumBBasal.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XlumBBasalss <- model.matrix(~ subtype, data=as.data.frame(model_lumBBasal))
fitlumBBasalss <- lmFit(as.data.frame(lumBBasal), XlumBBasalss)

fitlumBBasalss <- eBayes(fitlumBBasalss)
TTlumBBasalss <- topTable(fitlumBBasalss, number = dim(lumBBasal)[1])
pvalslumBBasalss <- fitlumBBasalss$p.value
fdradjusted_pvalslumBBasalINTERss <- p.adjust(pvalslumBBasalss[,1], method = "fdr")
fdradjusted_pvalslumBBasalsigINTERss <- fdradjusted_pvalslumBBasalINTERss[which(fdradjusted_pvalslumBBasalINTERss < 0.05)]
length(names(fdradjusted_pvalslumBBasalsigINTERss))
plotSA(fitlumBBasalss, main="Final model: Mean-variance trend")

treatedTT1lumBBasalss <- topTreat(fitlumBBasalss, coef=1, number=dim(lumBBasal)[1], lfc=1)
treatedTT1lumBBasalsigss <- treatedTT1lumBBasalss[which(treatedTT1lumBBasalss$adj.P.Val < 0.05),]
treatedTT1lumBBasalsigss
```

```{r}
plot(treatedTT1lumBBasalss$logFC, -log10(treatedTT1lumBBasalss$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT1lumBBasalsigss$logFC, -log10(treatedTT1lumBBasalsigss$adj.P.Val), col="red", pch="*")
```

plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4lumBBasal$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4lumBBasal$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4lumBBasal$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumBBasal$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumBBasal$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumBBasal$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumBBasal$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumBBasal$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumBBasal$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumBBasal$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumBBasals$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumBBasals$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumBBasals$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumBBasals$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumBBasals$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumBBasals$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumBBasalsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT2lumBBasals$adj.P.Val) / length(ntreatedTT2lumBBasals$adj.P.Val)), nconvSort2s, col="#cc0052", pch=17, xlab="Expected Adjusted p-value (-Log10)", ylab="Observed Adjusted p-value (-Log10)", las=1, bty="l")
#points(-log10(1:length(treatedTT2lumBBasals$adj.P.Val) / length(treatedTT2lumBBasals$adj.P.Val)), convSort2s, col="#cc0052", pch=17)
#points(-log10(1:length(treatedTT1lumBBasalss$adj.P.Val) / length(treatedTT1lumBBasalss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumBBasal$adj.P.Val) / length(ntreatedTT4lumBBasal$adj.P.Val)), nconvSort4, col="orange")
points(-log10(1:length(treatedTT2lumBBasal$adj.P.Val) / length(treatedTT2lumBBasal$adj.P.Val)), convSort2, col="green")
points(-log10(1:length(ntreatedTT2lumBBasal$adj.P.Val) / length(ntreatedTT2lumBBasal$adj.P.Val)), nconvSort2, col="#668cff")
#points(-log10(1:length(ntreatedTT3lumBBasal$adj.P.Val) / length(ntreatedTT3lumBBasal$adj.P.Val)), nconvSort3, col="#0033cc")
#points(-log10(1:length(ntreatedTT1lumBBasal$adj.P.Val) / length(ntreatedTT1lumBBasal$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumBBasal$adj.P.Val) / length(ntreatedTT1lumBBasal$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumBBasal$adj.P.Val) / length(ntreatedTT4lumBBasal$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumBBasal$adj.P.Val) / length(ntreatedTT2lumBBasal$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumBBasal$adj.P.Val) / length(ntreatedTT3lumBBasal$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumBBasal$adj.P.Val) / length(treatedTT1lumBBasal$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumBBasal$adj.P.Val) / length(treatedTT3lumBBasal$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumBBasal$adj.P.Val) / length(treatedTT4lumBBasal$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumBBasal$adj.P.Val) / length(treatedTT2lumBBasal$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2lumBBasal$adj.P.Val) / length(treatedTT2lumBBasal$adj.P.Val)), convSort2, xlab="Expected Adjusted p-value (-Log10)", ylab="Observed Adjusted p-value (-Log10)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumBBasal$adj.P.Val) / length(ntreatedTT2lumBBasal$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumBBasal$adj.P.Val) / length(treatedTT3lumBBasal$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumBBasal$adj.P.Val) / length(ntreatedTT3lumBBasal$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumBBasal$adj.P.Val) / length(treatedTT4lumBBasal$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumBBasal$adj.P.Val) / length(ntreatedTT4lumBBasal$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```













