---
title: "lumBNormal"
output: html_document
---

```{r, include = FALSE} 
library(ggplot2)#7 rows
library(EDASeq)
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(dplyr)
library(SummarizedExperiment)
library(limma)
library(edgeR)
library(jtools) # for summ()
library(MASS)
library(splines)
library(tidyverse)
library(ggplot2)
library(interactions)
library("readxl")
library(preprocessCore)
library(sjPlot)
library(DESeq2)
library(sjlabelled)
library(sjmisc)
library(ggplot2)
library(NOISeq)
library(tidyverse)
library(fgsea)
library(org.Hs.eg.db)
library(dplyr)
library(gage)
library(fgsea)
library(ggplot2)
library(GSVAdata)
library(org.Hs.eg.db)
library(GSEABase)
library("ggsci")
library("gridExtra")
library(ensembldb)
library(EnsDb.Hsapiens.v86)
```

```{r }

load("~/Documents/project/normaliseddataCPE.Rdata")
load("~/Documents/project/modelmatricesCPE.Rdata")

#load(file="~/Documents/project/normaliseddata.Rdata")
lumBNormal <- datalumBNormal
#load(file="~/Documents/project/modelmatrices.Rdata")
```



read https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5354413/#!po=38.0000 paper. It is very similar to what we are doing here

the MAIN model:
```{r}
XlumBNormal <- model.matrix(~ subtype * CPE, data=as.data.frame(model_lumBNormal))
fitlumBNormal <- lmFit(as.data.frame(lumBNormal), XlumBNormal)

#contr.matrix <- makeContrasts(SubvsCPE=subtype-CPE, Sub_CPE=subtype:CPE, levels = colnames(XlumBNormal))
#contr.matrix
#contrastsfitlumBNormal <- contrasts.fit(fitlumBNormal, contrasts=contr.matrix)
#contrastsfitlumBNormal <- eBayes(contrastsfitlumBNormal)
fitlumBNormal <- eBayes(fitlumBNormal)
TTlumBNormal <- topTable(fitlumBNormal, number = dim(lumBNormal)[1])
pvalslumBNormal <- fitlumBNormal$p.value
fdradjusted_pvalslumBNormal <- p.adjust(pvalslumBNormal[,4], method = "fdr")
fdradjusted_pvalslumBNormalsig <- fdradjusted_pvalslumBNormal[which(fdradjusted_pvalslumBNormal < 0.05)]
names(fdradjusted_pvalslumBNormalsig)
top95lumBNormal <- TTlumBNormal[which(rownames(TTlumBNormal) %in% names(fdradjusted_pvalslumBNormalsig)),]
#plotMA(treatedTT4lumBNormal , alpha = 0.05,  main = "WT vs. SNF2  mutants", ylim = c(-4,4))
#hist(TTlumBNormal$adj.P.Val ,col = "grey", border = "white", xlab = "", main = "frequencies  of p-values")
plotSA(fitlumBNormal, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumBNormal)
treatedTT1lumBNormal <- topTreat(fitlumBNormal, coef=1, number=dim(lumBNormal)[1], lfc=1)
treatedTT1lumBNormalsig <- treatedTT1lumBNormal[which(treatedTT1lumBNormal$adj.P.Val < 0.05),]
treatedTT1lumBNormalsig
treatedTT2lumBNormal <- topTreat(fitlumBNormal, coef=2, number=dim(lumBNormal)[1], lfc=1)
treatedTT2lumBNormalsig <- treatedTT2lumBNormal[which(treatedTT2lumBNormal$adj.P.Val < 0.05),]
treatedTT2lumBNormalsig
treatedTT3lumBNormal <- topTreat(fitlumBNormal, coef=3, number=dim(lumBNormal)[1], lfc=1)
treatedTT3lumBNormalsig <- treatedTT3lumBNormal[which(treatedTT3lumBNormal$adj.P.Val < 0.05),]
treatedTT3lumBNormalsig
treatedTT4lumBNormal <- topTreat(fitlumBNormal, coef=4, number=dim(lumBNormal)[1], lfc=1)
treatedTT4lumBNormalsig <- treatedTT4lumBNormal[which(treatedTT4lumBNormal$adj.P.Val < 0.05),]
treatedTT4lumBNormalsig
#plot(density(treatedTT2lumBNormalsig$logFC))
#ENSG00000122566
#ENSG00000122566

save(treatedTT4lumBNormalsig, treatedTT2lumBNormalsig, file="lumBNormaloutputCPE.Rdata")
```






```{r}
plot(treatedTT4lumBNormal$logFC, -log10(treatedTT4lumBNormal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumBNormalsig$logFC, -log10(treatedTT4lumBNormalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumBNormal <-model_lumBNormal
nmodel_lumBNormal$CPE <- (1- nmodel_lumBNormal$CPE)
nXlumBNormal <- model.matrix(~ subtype * CPE, data=as.data.frame(nmodel_lumBNormal))
nfitlumBNormal <- lmFit(as.data.frame(lumBNormal), nXlumBNormal)

nfitlumBNormal <- eBayes(nfitlumBNormal)
nTTlumBNormal<- topTable(nfitlumBNormal, number = dim(lumBNormal)[1])
npvalslumBNormal <- nfitlumBNormal$p.value
nfdradjusted_pvalslumBNormal <- p.adjust(npvalslumBNormal[,4], method = "fdr")
nfdradjusted_pvalslumBNormalsig <- nfdradjusted_pvalslumBNormal[which(nfdradjusted_pvalslumBNormal < 0.05)]
names(nfdradjusted_pvalslumBNormalsig)
ntop95lumBNormal <- nTTlumBNormal[which(rownames(nTTlumBNormal) %in% names(nfdradjusted_pvalslumBNormalsig)),]

ntreatedTT1lumBNormal <- topTreat(nfitlumBNormal, coef=1, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT1lumBNormalsig <- ntreatedTT1lumBNormal[which(ntreatedTT1lumBNormal$adj.P.Val < 0.05),]
ntreatedTT1lumBNormalsig
ntreatedTT2lumBNormal <- topTreat(nfitlumBNormal, coef=2, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT2lumBNormalsig <- ntreatedTT2lumBNormal[which(ntreatedTT2lumBNormal$adj.P.Val < 0.05),]
ntreatedTT2lumBNormalsig
ntreatedTT3lumBNormal <- topTreat(nfitlumBNormal, coef=3, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT3lumBNormalsig <- ntreatedTT3lumBNormal[which(ntreatedTT3lumBNormal$adj.P.Val < 0.05),]
ntreatedTT3lumBNormalsig
ntreatedTT4lumBNormal <- topTreat(nfitlumBNormal, coef=4, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT4lumBNormalsig <- ntreatedTT4lumBNormal[which(ntreatedTT4lumBNormal$adj.P.Val < 0.05),]
ntreatedTT4lumBNormalsig
```

single linear model for 1-tumour purity:
```{r}
nCPEmodel_lumBNormal <- as.data.frame(model_lumBNormal)
nCPEmodel_lumBNormal$CPE <- (nCPEmodel_lumBNormal$CPE)
nthelm2lumBNormal <- lm(lumBNormal[17876,] ~ subtype * CPE, data=nCPEmodel_lumBNormal)

plot_model(nthelm2lumBNormal, type = "pred", terms = c( "CPE","subtype"))
which(rownames(lumBNormal) == "ENSG00000214688") # this is a gene responsible for ER expression
```

```{r}
normallumBNormal <- treatedTT2lumBNormalsig[which(rownames(treatedTT2lumBNormalsig) %in% rownames(treatedTT4lumBNormalsig)),]
normallumBNormal <- normallumBNormal[which(!rownames(normallumBNormal) %in% rownames(ntreatedTT2lumBNormalsig)),]

cancerlumBNormal <- ntreatedTT2lumBNormalsig[which(rownames(ntreatedTT2lumBNormalsig) %in% rownames(ntreatedTT4lumBNormalsig)),]
cancerlumBNormal <- cancerlumBNormal[which(!rownames(cancerlumBNormal) %in% rownames(treatedTT2lumBNormalsig)),]
cancerlumBNormal
normallumBNormal 

save(cancerlumBNormal, normallumBNormal, file="lumBNormal_C_N_CPE.Rdata")
```

```{r}

mypal = pal_d3("category10", alpha = 0.7)(9)
cb3 <-mypal[c(1,4,3)]
treatedTT4lumBNormalsig$status <- ifelse(rownames(treatedTT4lumBNormalsig) %in% rownames(cancerlumBNormal), "Within Cancer Cells", ifelse(rownames(treatedTT4lumBNormalsig) %in% rownames(normallumBNormal), "Within Normal Cells", "Both"))
treatedTT4lumBNormalsig$colour <- ifelse(rownames(treatedTT4lumBNormalsig) %in% rownames(cancerlumBNormal), cb3[2], ifelse(rownames(treatedTT4lumBNormalsig) %in% rownames(normallumBNormal), cb3[3], cb3[1]))
geneslumBNormal <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT4lumBNormalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
geneslumBNormal <- geneslumBNormal[!duplicated(geneslumBNormal$GENEID),]
geneslumBNormal$logFC <- treatedTT4lumBNormalsig[which(rownames(treatedTT4lumBNormalsig) %in% geneslumBNormal$GENEID),][geneslumBNormal$GENEID,]$logFC
geneslumBNormal$status <- treatedTT4lumBNormalsig[which(rownames(treatedTT4lumBNormalsig) %in% geneslumBNormal$GENEID),][geneslumBNormal$GENEID,]$status
lumBNormalsigs <- ggplot(geneslumBNormal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal B vs Normal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb3)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("lumBNormalsigsCPE.png")



mypal = pal_d3("category10", alpha = 0.7)(9)
cb2 <-mypal[c(4,3)]
cancernormalonlylumBNormal <-geneslumBNormal[which(geneslumBNormal$status != "Both"),]
cancernormalonlylumBNormalgraph <- ggplot(cancernormalonlylumBNormal, (aes(x = reorder(SYMBOL, -logFC), y = logFC, fill=status))) + 
  geom_col(width=0.7) +
   ggtitle("Luminal B vs Normal-like Significantly Differentially Expressed Genes") +
  scale_y_continuous(name = "Log Fold Change") +
  scale_fill_manual(values=cb2)+theme_bw()+
    theme(
      axis.line.y = element_blank(),
      axis.ticks.y = element_blank(),
            plot.title = element_text(size = 11.5, face="bold"),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      #axis.text.x = element_text(size = 6, angle = 60, vjust = 1, hjust = 1),
      axis.title.x = element_blank(),
      plot.margin = margin(6, 6, 3, 3),
      panel.background=element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.major.y = element_line(size = 0.2, linetype = 'solid',colour = "white"),
      panel.ontop = TRUE, 
      legend.position= c(0.6, 0.92), legend.direction="horizontal",
      legend.text = element_text(size = 6), # legend text  was a little large
      legend.key.size = unit(0.6, "lines"),
      legend.title = element_blank())# legend keys were a little large

ggsave("cancernormalonlylumBNormalCPE.png")
```

Simple linear model NO interaciton:

the MAIN model:
```{r}
XlumBNormals <- model.matrix(~ subtype + CPE, data=as.data.frame(model_lumBNormal))
fitlumBNormals <- lmFit(as.data.frame(lumBNormal), XlumBNormals)
fitlumBNormals <- eBayes(fitlumBNormals)
TTlumBNormals <- topTable(fitlumBNormals, number = dim(lumBNormal)[1])
pvalslumBNormals <- fitlumBNormals$p.value
fdradjusted_pvalslumBNormalINTERs <- p.adjust(pvalslumBNormals[,1], method = "fdr")
fdradjusted_pvalslumBNormalsigINTERs <- fdradjusted_pvalslumBNormalINTERs[which(fdradjusted_pvalslumBNormalINTERs < 0.05)]
length(names(fdradjusted_pvalslumBNormalsigINTERs))
fdradjusted_pvalslumBNormalSUBs <- p.adjust(pvalslumBNormals[,2], method = "fdr")
fdradjusted_pvalslumBNormalsigSUBs <- fdradjusted_pvalslumBNormalSUBs[which(fdradjusted_pvalslumBNormalSUBs < 0.05)]
length(names(fdradjusted_pvalslumBNormalsigSUBs))
fdradjusted_pvalslumBNormalCPEs <- p.adjust(pvalslumBNormals[,3], method = "fdr")
fdradjusted_pvalslumBNormalsigCPEs <- fdradjusted_pvalslumBNormalCPEs[which(fdradjusted_pvalslumBNormalCPEs < 0.05)]
length(names(fdradjusted_pvalslumBNormalsigCPEs))
plotSA(fitlumBNormals, main="Final model: Mean-variance trend")
hist(fdradjusted_pvalslumBNormalSUBs)

treatedTT1lumBNormals <- topTreat(fitlumBNormals, coef=1, number=dim(lumBNormal)[1], lfc=1)
treatedTT1lumBNormalsigs <- treatedTT1lumBNormals[which(treatedTT1lumBNormals$adj.P.Val < 0.05),]
treatedTT1lumBNormalsigs
treatedTT2lumBNormals <- topTreat(fitlumBNormals, coef=2, number=dim(lumBNormal)[1], lfc=1)
treatedTT2lumBNormalsigs <- treatedTT2lumBNormals[which(treatedTT2lumBNormals$adj.P.Val < 0.05),]
treatedTT2lumBNormalsigs
treatedTT3lumBNormals <- topTreat(fitlumBNormals, coef=3, number=dim(lumBNormal)[1], lfc=1)
treatedTT3lumBNormalsigs <- treatedTT3lumBNormals[which(treatedTT3lumBNormals$adj.P.Val < 0.05),]
treatedTT3lumBNormalsigs


treatedTT2lumBNormalsigs[which(rownames(treatedTT2lumBNormalsigs) %in% rownames(treatedTT4lumBNormalsig)),]
treatedTT2lumBNormalsigs[which(rownames(treatedTT2lumBNormalsigs) %in% rownames(cancerlumBNormal)),]
treatedTT2lumBNormalsigs[which(rownames(treatedTT2lumBNormalsigs) %in% rownames(normallumBNormal)),]
```

```{r}
plot(treatedTT4lumBNormal$logFC, -log10(treatedTT4lumBNormal$adj.P.Val), pch="*", xlab="Log2 Fold Change", ylab="-10log (adjusted p-value)") 
#first plot -10*log of adjusted p-values vs log Fold Change values for each of the genes that were differentially expressed

abline(h=-log10(0.05), v=c(-1, 1), col="red", lty=2) 
#use the abline to highlight the points that pass our thresholds of adjusted p-value being equal or less than 0.05 and absolute value of log Fold Change being bigger than 1.

points(treatedTT4lumBNormalsig$logFC, -log10(treatedTT4lumBNormalsig$adj.P.Val), col="red", pch="*")
```

plot for 1-tumour purity linear models:

```{r}
nmodel_lumBNormals <- model_lumBNormal
nmodel_lumBNormals$CPE <- (1- nmodel_lumBNormals$CPE)
nXlumBNormals <- model.matrix(~ subtype + CPE, data=as.data.frame(nmodel_lumBNormals))
nfitlumBNormals <- lmFit(as.data.frame(lumBNormal), nXlumBNormals)
nfitlumBNormals <- eBayes(nfitlumBNormals)
nTTlumBNormals<- topTable(nfitlumBNormals, number = dim(lumBNormal)[1])
npvalslumBNormals <- nfitlumBNormals$p.value
ntreatedTT1lumBNormals <- topTreat(nfitlumBNormals, coef=1, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT1lumBNormalsigs <- ntreatedTT1lumBNormals[which(ntreatedTT1lumBNormals$adj.P.Val < 0.05),]
ntreatedTT1lumBNormalsigs
ntreatedTT2lumBNormals <- topTreat(nfitlumBNormals, coef=2, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT2lumBNormalsigs <- ntreatedTT2lumBNormals[which(ntreatedTT2lumBNormals$adj.P.Val < 0.05),]
ntreatedTT2lumBNormalsigs
#ntreatedTT2lumBNormalsigs[which(rownames(ntreatedTT2lumBNormalsigs) %in% rownames(normallumBNormal)),]
ntreatedTT3lumBNormals <- topTreat(nfitlumBNormals, coef=3, number=dim(lumBNormal)[1], lfc=1)
ntreatedTT3lumBNormalsigs <- ntreatedTT3lumBNormals[which(ntreatedTT3lumBNormals$adj.P.Val < 0.05),]
ntreatedTT3lumBNormalsigs
```





merged linear plot for top10:
```{r}
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(xtable)
Subtype_names <- c("Luminal B", "Normal-like")
CPEmodel_lumBNormal <- as.data.frame(model_lumBNormal)
genes <- ensembldb::select(EnsDb.Hsapiens.v86, keys= rownames(treatedTT1lumBNormalsig), keytype =   "GENEID", columns = c(  "GENEID", "SYMBOL", "ENTREZID"))
genes<- genes[!duplicated(genes$GENEID),]

top10Genes <- genes[which(genes$GENEID %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
top10Genes <- top10Genes[order(top10Genes$GENEID),]
top10GeneslumBNormal <- lumBNormal[which(rownames(lumBNormal) %in% c("ENSG00000141510", "ENSG00000121879", "ENSG00000039068", "ENSG00000107485", "ENSG00000055609", "ENSG00000095015","ENSG00000171862", "ENSG00000141027" , "ENSG00000159216", "ENSG00000117713")),]
#top10GeneslumBNormal[top10Genes$GENEID,]
top10GeneslumBNormal <- top10GeneslumBNormal[order(rownames(top10GeneslumBNormal)),]
dim(top10GeneslumBNormal)
all.equal(rownames(top10GeneslumBNormal), top10Genes$GENEID)
rownames(top10GeneslumBNormal) <- top10Genes$SYMBOL
top10lmlumBNormal <-  numeric(length = nrow(top10GeneslumBNormal))
plotlumBNormalCPE1 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[1,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[1])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(color = "red",face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),legend.position = c(0.05, 0.05),
    legend.justification = c("left", "bottom"),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE1, file=paste0("~/Documents/project/plots/lumBNormalCPE_111.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBNormalCPE2 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[2,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[2])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE2, file=paste0("~/Documents/project/plots/lumBNormalCPE_211.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBNormalCPE3 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[3,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[3])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE3, file=paste0("~/Documents/project/plots/lumBNormalCPE_311.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBNormalCPE4 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[4,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[4])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE4, file=paste0("~/Documents/project/plots/lumBNormalCPE_411.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBNormalCPE5 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[5,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[5])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE5, file=paste0("~/Documents/project/plots/lumBNormalCPE_511.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBNormalCPE6 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[6,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[6])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE6, file=paste0("~/Documents/project/plots/lumBNormalCPE_611.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBNormalCPE7 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[7,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[7])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE7, file=paste0("~/Documents/project/plots/lumBNormalCPE_711.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBNormalCPE8 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[8,],x=CPE,color=factor(subtype),shape=factor(subtype) ))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[8])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE8, file=paste0("~/Documents/project/plots/lumBNormalCPE_811.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)

plotlumBNormalCPE9 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[9,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
   
     ggtitle(rownames(top10GeneslumBNormal)[9])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE9, file=paste0("~/Documents/project/plots/lumBNormalCPE_911.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


plotlumBNormalCPE10 <- ggplot(CPEmodel_lumBNormal ,aes(y=top10GeneslumBNormal[10,],x=CPE,color=factor(subtype), shape=factor(subtype)))+
    scale_colour_manual(values = c("green4", "blue"), name = "Subtype", labels=Subtype_names) +
    scale_shape_manual(values = c(8, 1), name = "Subtype", labels=Subtype_names) +  xlab("Tumour Purity (CPE)") + # new label for x axis
     ggtitle(rownames(top10GeneslumBNormal)[10])+
  ylab ("Gene Expression") + theme_classic() +
  geom_point()+stat_smooth(method="lm",se=FALSE)+  theme (axis.text = element_text(size =10),
  panel.grid.major.x = element_line(size = 0.25, linetype = 'solid',colour = "lightgrey"), 
  panel.grid.minor.x = element_line(size = 0.1, linetype = 'solid',colour = "lightgrey"),
  axis.title = element_text(size =12),
  legend.title = element_text(face="bold",size=14),
  axis.text.y = element_blank(),
  axis.title.y=element_blank(),
  axis.ticks = element_blank(),
  legend.text = element_text(face="italic",size=10),
  plot.title = element_text(face="bold",size=14))
ggsave(plotlumBNormalCPE10, file=paste0("~/Documents/project/plots/lumBNormalCPE_1011.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)


mergedCPElumBNormal <-cowplot::plot_grid(plotlumBNormalCPE1, plotlumBNormalCPE2 + theme(legend.position="none"), plotlumBNormalCPE3 + theme(legend.position="none"),plotlumBNormalCPE4 + theme(legend.position="none"), plotlumBNormalCPE5 + theme(legend.position="none"), plotlumBNormalCPE6 + theme(legend.position="none"),plotlumBNormalCPE7 + theme(legend.position="none"), plotlumBNormalCPE8 + theme(legend.position="none"), plotlumBNormalCPE9 + theme(legend.position="none"),plotlumBNormalCPE10 + theme(legend.position="none"), align = "hv", ncol=5)

ggsave(mergedCPElumBNormal, file=paste0("~/Documents/project/plots/mergedCPElumBNormal.png"), width = 44.45, height = 27.78, units = "cm", dpi=300)
```

NO tumour purity effect taken into account:

the MAIN model:
```{r}
XlumBNormalss <- model.matrix(~ subtype, data=as.data.frame(model_lumBNormal))
fitlumBNormalss <- lmFit(as.data.frame(lumBNormal), XlumBNormalss)

fitlumBNormalss <- eBayes(fitlumBNormalss)
TTlumBNormalss <- topTable(fitlumBNormalss, number = dim(lumBNormal)[1])
pvalslumBNormalss <- fitlumBNormalss$p.value
fdradjusted_pvalslumBNormalINTERss <- p.adjust(pvalslumBNormalss[,1], method = "fdr")
fdradjusted_pvalslumBNormalsigINTERss <- fdradjusted_pvalslumBNormalINTERss[which(fdradjusted_pvalslumBNormalINTERss < 0.05)]
length(names(fdradjusted_pvalslumBNormalsigINTERss))
plotSA(fitlumBNormalss, main="Final model: Mean-variance trend")

treatedTT1lumBNormalss <- topTreat(fitlumBNormalss, coef=1, number=dim(lumBNormal)[1], lfc=1)
treatedTT1lumBNormalsigss <- treatedTT1lumBNormalss[which(treatedTT1lumBNormalss$adj.P.Val < 0.05),]
treatedTT1lumBNormalsigss
```



plot QQ plots adj p values:
```{r}
#qqnorm(ntreatedTT4lumBNormal$adj.P.Val, pch = 1, frame = FALSE)
#qqline(ntreatedTT4lumBNormal$adj.P.Val, col = "steelblue", lwd = 2)

#' QQ-plot of the simulated P-values for bulk, cancer model and random CPE model.
nconvSort4 <- sort(-log10(ntreatedTT4lumBNormal$adj.P.Val), decreasing=T)
nconvSort3 <- sort(-log10(ntreatedTT3lumBNormal$adj.P.Val), decreasing=T)
nconvSort2 <- sort(-log10(ntreatedTT2lumBNormal$adj.P.Val), decreasing=T)
nconvSort1 <- sort(-log10(ntreatedTT1lumBNormal$adj.P.Val), decreasing=T)
convSort4 <- sort(-log10(treatedTT4lumBNormal$adj.P.Val), decreasing=T)
convSort3 <- sort(-log10(treatedTT3lumBNormal$adj.P.Val), decreasing=T)
convSort2 <- sort(-log10(treatedTT2lumBNormal$adj.P.Val), decreasing=T)
convSort1 <- sort(-log10(treatedTT1lumBNormal$adj.P.Val), decreasing=T)

nconvSort1s <- sort(-log10(ntreatedTT1lumBNormals$adj.P.Val), decreasing=T)
convSort1s <- sort(-log10(treatedTT1lumBNormals$adj.P.Val), decreasing=T)
nconvSort2s <- sort(-log10(ntreatedTT2lumBNormals$adj.P.Val), decreasing=T)
convSort2s <- sort(-log10(treatedTT2lumBNormals$adj.P.Val), decreasing=T)
nconvSort3s <- sort(-log10(ntreatedTT3lumBNormals$adj.P.Val), decreasing=T)
convSort3s <- sort(-log10(treatedTT3lumBNormals$adj.P.Val), decreasing=T)

convSort1ss <- sort(-log10(treatedTT1lumBNormalsigss$adj.P.Val), decreasing=T)
#' Make a QQ plot 
plot(-log10(1:length(ntreatedTT1lumBNormals$adj.P.Val) / length(ntreatedTT1lumBNormals$adj.P.Val)), nconvSort1s, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l",  pch=17)
points(-log10(1:length(ntreatedTT2lumBNormals$adj.P.Val) / length(ntreatedTT2lumBNormals$adj.P.Val)), nconvSort2s, col="#ff99c2", pch=17)
points(-log10(1:length(ntreatedTT3lumBNormals$adj.P.Val) / length(ntreatedTT3lumBNormals$adj.P.Val)), nconvSort3s, col="#cc0052", pch=17)
points(-log10(1:length(treatedTT1lumBNormalss$adj.P.Val) / length(treatedTT1lumBNormalss$adj.P.Val)), convSort1ss, col="orange", pch=1)
points(-log10(1:length(ntreatedTT4lumBNormal$adj.P.Val) / length(ntreatedTT4lumBNormal$adj.P.Val)), nconvSort4, col="#ccd9ff")
points(-log10(1:length(ntreatedTT2lumBNormal$adj.P.Val) / length(ntreatedTT2lumBNormal$adj.P.Val)), nconvSort2, col="#668cff")
points(-log10(1:length(ntreatedTT3lumBNormal$adj.P.Val) / length(ntreatedTT3lumBNormal$adj.P.Val)), nconvSort3, col="#0033cc")
points(-log10(1:length(ntreatedTT1lumBNormal$adj.P.Val) / length(ntreatedTT1lumBNormal$adj.P.Val)), nconvSort1, col="#002699")
abline(0,1, col="red")

plot(-log10(1:length(ntreatedTT1lumBNormal$adj.P.Val) / length(ntreatedTT1lumBNormal$adj.P.Val)), nconvSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumBNormal$adj.P.Val) / length(ntreatedTT4lumBNormal$adj.P.Val)), nconvSort4, col="#ff4d94")
points(-log10(1:length(ntreatedTT2lumBNormal$adj.P.Val) / length(ntreatedTT2lumBNormal$adj.P.Val)), nconvSort2, col="#ff99c2")
points(-log10(1:length(ntreatedTT3lumBNormal$adj.P.Val) / length(ntreatedTT3lumBNormal$adj.P.Val)), nconvSort3, col="#cc0052")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT1lumBNormal$adj.P.Val) / length(treatedTT1lumBNormal$adj.P.Val)), convSort1, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(treatedTT3lumBNormal$adj.P.Val) / length(treatedTT3lumBNormal$adj.P.Val)), convSort3, col="#ff4d94")
points(-log10(1:length(treatedTT4lumBNormal$adj.P.Val) / length(treatedTT4lumBNormal$adj.P.Val)), convSort4, col="#ff99c2")
points(-log10(1:length(treatedTT2lumBNormal$adj.P.Val) / length(treatedTT2lumBNormal$adj.P.Val)), convSort2, col="#cc0052")
abline(0,1, col="red")


plot(-log10(1:length(treatedTT2lumBNormal$adj.P.Val) / length(treatedTT2lumBNormal$adj.P.Val)), convSort2, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="orange", bty="l")
points(-log10(1:length(ntreatedTT2lumBNormal$adj.P.Val) / length(ntreatedTT2lumBNormal$adj.P.Val)), nconvSort2, col="#809fff")
abline(0,1, col="red")

plot(-log10(1:length(treatedTT3lumBNormal$adj.P.Val) / length(treatedTT3lumBNormal$adj.P.Val)), convSort3, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT3lumBNormal$adj.P.Val) / length(ntreatedTT3lumBNormal$adj.P.Val)), nconvSort3, col="#0040ff", pch=4)
abline(0,1, col="red")

plot(-log10(1:length(treatedTT4lumBNormal$adj.P.Val) / length(treatedTT4lumBNormal$adj.P.Val)), convSort4, xlab="-Log10 P-Value (uniform distribution)", ylab="-Log10 P-Value (measured)", las=1, col="#ff0066", bty="l")
points(-log10(1:length(ntreatedTT4lumBNormal$adj.P.Val) / length(ntreatedTT4lumBNormal$adj.P.Val)), nconvSort4, col="#0040ff", pch=4)
abline(0,1, col="red")
```

























